<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>About</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/readable.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">PSYMETAB</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="setup.html">Setup</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/jennysjaarda/PSYMETAB">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">About</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2019-12-06
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>PSYMETAB/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.5.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20191126code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20191126)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20191126code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20191126)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomjennysjaardaPSYMETABtree5e454c33eb2720be586b82acfc4be3c4c52c9124targetblank5e454c3a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/jennysjaarda/PSYMETAB/tree/5e454c33eb2720be586b82acfc4be3c4c52c9124" target="_blank">5e454c3</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomjennysjaardaPSYMETABtree5e454c33eb2720be586b82acfc4be3c4c52c9124targetblank5e454c3a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .drake/
    Ignored:    analysis/QC/
    Ignored:    data/processed/
    Ignored:    data/raw/

Untracked files:
    Untracked:  post_impute_qc.out

Unstaged changes:
    Deleted:    post_imputation_qc.log
    Deleted:    pre_imputation_qc.log
    Modified:   pre_impute_qc.out
    Deleted:    qc_part2.out

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the R Markdown and HTML files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view them.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jennysjaarda/PSYMETAB/97b2b48ab966ae123bccd64aacf2e0d331952d2e/docs/about.html" target="_blank">97b2b48</a>
</td>
<td>
Jenny
</td>
<td>
2019-11-26
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jennysjaarda/PSYMETAB/blob/2db9bcfcc7febefb29ff93ea2a30e41c0f0156f2/analysis/about.Rmd" target="_blank">2db9bcf</a>
</td>
<td>
Jenny
</td>
<td>
2019-11-26
</td>
<td>
wflow_publish(all = T)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jennysjaarda/PSYMETAB/2db9bcfcc7febefb29ff93ea2a30e41c0f0156f2/docs/about.html" target="_blank">2db9bcf</a>
</td>
<td>
Jenny
</td>
<td>
2019-11-26
</td>
<td>
wflow_publish(all = T)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jennysjaarda/PSYMETAB/5fb939e5f8209480d44c3c98625a40ff85c59e44/docs/about.html" target="_blank">5fb939e</a>
</td>
<td>
Jenny Sjaarda
</td>
<td>
2019-11-26
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jennysjaarda/PSYMETAB/blob/7bae873483c621107e61ecd6592546ed1aef112a/analysis/about.Rmd" target="_blank">7bae873</a>
</td>
<td>
Jenny Sjaarda
</td>
<td>
2019-11-26
</td>
<td>
Start workflowr project.
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p>Describe your project.</p>
<p>see <code>data/README.md</code> for information on data inputs see <code>scripst/README.md</code> for information on scripts</p>
<ul>
<li><a href="info.html">INFO</a></li>
</ul>
<p><strong>Last updated:</strong> 2019-12-06</p>
<p><strong>Code version:</strong> 5e454c33eb2720be586b82acfc4be3c4c52c9124</p>
<p>We used this site to collaborate and share our results. Please feel free to explore. The results that made it into the final paper are in the section <a href="#finalizing">Finalizing</a> below. Here are some useful links:</p>
<ul>
<li><a href="http://www.nature.com/articles/srep39921">Batch effects and the effective design of single-cell gene expression studies</a> (published in 2017) - <a href="http://biorxiv.org/content/early/2016/07/08/062919">Pre-print</a> (posted in 2016)</li>
<li>GEO record <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE77288">GSE77288</a> for the raw FASTQ files</li>
<li><a href="https://github.com/jennysjaarda/PSYMETAB">GitHub repo</a> for code and data</li>
</ul>
<div id="setting-up-the-project" class="section level2">
<h2>Setting up the project</h2>
</div>
<div id="finalizing" class="section level2">
<h2>Finalizing</h2>
<ul>
<li><a href="process-samples.html">Process sequencing data - From fastq to gene counts</a></li>
<li><a href="prepare-counts.html">Create gene-x-sample count matrices and annotation files</a></li>
<li><a href="qc-filter-ipsc.html">Quality control of single cells</a></li>
<li><a href="filter.html">Filter genes and low-quality single cells</a></li>
<li><a href="qc-plots.html">Quality control of UMIs and ERCC spike-ins</a></li>
<li><a href="compare-reads-v-molecules-per-batch-test.html">Compare reads to molecules counts</a></li>
<li><a href="subsample-final.html">Subsampling</a> (<a href="subsample-lcl-final.html">LCL</a>, <a href="subsample-final-rep.html">per batch</a>)</li>
<li><a href="cpm.html">Standardize with counts per million</a></li>
<li><a href="poisson-glm-normalization.html">Account for technical noise measured with ERCC controls using Poisson GLM transformation</a></li>
<li><a href="batch-correction.html">Mixed effect model for batch correction</a></li>
<li><a href="data-transformation.html">Visualization of data transformations</a></li>
<li><a href="cv-adjusted-summary-pois-final.html">Comparing cell-to-cell variation between individuals</a> (<a href="paper-additional-analysis.html">Additional figures</a>)</li>
</ul>
</div>
<div id="process-sequence-data" class="section level2">
<h2>Process sequence data</h2>
<ul>
<li><a href="process-samples.html">Pipeline</a> - (<a href="process-lcl.html">LCLs</a>)</li>
<li><a href="total-counts.html">Total counts</a></li>
<li><a href="qc-by-lane.html">Quality control at level of sequencing lane</a></li>
<li><a href="sum-counts-per-sample.html">Sum counts per sample</a></li>
<li><a href="qc-cell-ipsc.html">Quality control at level of single cell samples</a> - (<a href="qc-cell-lcl.html">LCLs</a>)
<ul>
<li><a href="depth.html">Sequencing depth and cellular RNA content</a> - (<a href="depth-final.html">Final</a>)</li>
<li><a href="qc-filter-ipsc.html">Attempt to finalize quality control</a></li>
</ul></li>
</ul>
</div>
<div id="analysis" class="section level2">
<h2>Analysis</h2>
<ul>
<li><a href="noisy-gene.html">Identification of noisy genes</a></li>
<li><a href="compare-reads-v-molecules.html">Compare read versus molecule counts</a> - (<a href="compare-reads-v-molecules-per-batch.html">per batch</a>,<a href="compare-reads-v-molecules-per-batch-test.html">per batch conversion rates comparison</a>, <a href="compare-reads-v-molecules-in-cells.html">per cell</a>)
<ul>
<li><a href="islam2014.html">Islam et al., 2014</a></li>
<li><a href="vallejos2015.html">Vallejos et al., 2015</a></li>
</ul></li>
<li><a href="variance.html">Variance within and between individaul</a></li>
<li><a href="ercc-correlation.html">Correlation with ERCC spike-ins</a>
<ul>
<li><a href="ercc-counts.html">Counts of each ERCC in single cell</a></li>
<li><a href="capture-efficiency.html">Capture efficiency</a></li>
<li><a href="ercc-over-sequenced.html">Over-sequenced ERCC in LCLs</a></li>
<li><a href="ercc-cpm.html">cmp ERCC or not</a></li>
</ul></li>
<li><a href="ercc-normalization.html">ERCC normalization</a>
<ul>
<li><a href="gamma-regression-model.html">Gamma regression model (GRM)</a></li>
<li><a href="basics.html">BASiCS</a>
<ul>
<li><a href="basics-wo-batch.html">without modeling batch</a></li>
<li>Using too few iterations gives a poor fit. These are older versions of the analysis kept to demonstrate this(<a href="basics-4000.html">4000 iterations</a>, <a href="basics-20000.html">20000 iterations</a>)</li>
</ul></li>
<li><a href="shrunk-linear-transform.html">Shrunk linear transformation</a></li>
<li><a href="basics-variance-components.html">Variance components</a></li>
<li><a href="poisson-glm-normalization.html">Poisson GLM normalization</a></li>
</ul></li>
<li><a href="subsample.html">Subsample</a> - (<a href="subsample-lcl.html">LCLs</a>)
<ul>
<li><a href="correlate-single-to-bulk.html">Correlation of single and bulk cells</a></li>
<li><a href="detect-genes.html">Number of expressed genes</a> - (<a href="detect-genes-lcl.html">LCLs</a>)</li>
<li><a href="subsample-high-coverage-lcl.html">High coverage single cells</a></li>
<li><a href="subsample-variance.html">Cell to cell variance</a></li>
</ul></li>
<li><a href="cell-cycle.html">Cell-cycle analysis</a> - (<a href="cell-cycle-post-correction.html">final</a>), (<a href="cell-cycle-investigate.html">cycle-gene-set</a>)</li>
<li><a href="stemness.html">pluripotency gene expression</a></li>
<li><a href="pca-correlation-rafa.html">Proportion of detected genes</a> - (<a href="pca-correlation-rafa-filter.html">filter and final</a>)
<ul>
<li><a href="gene-and-total-molecule.html">Detected genes and total molecule counts</a></li>
<li><a href="prop-genes-detected.html">More exploring of the proportion of genes detected</a> - <a href="prop-genes-detected-unfiltered.html">Using all cells</a></li>
</ul></li>
<li><a href="batch-limma.html">Batch effect correction with mixed modeling</a> - (<a href="batch-limma-final.html">linear transformation, filter and final</a>, <a href="batch-limma-poisson.html">Poisson transformation, filter and final</a>)
<ul>
<li><a href="batch-limma-final-per-gene.html">Per gene correction (filtered and final)</a></li>
</ul></li>
<li><a href="pca-individual.html">Individual PCAs</a></li>
<li>Ordering effect of capture sites
<ul>
<li>All genes (<a href="oefinder-analysis.html">DESeq-normalized</a>, <a href="oefinder-analysis-cpm.html">CPM-normalized</a>)</li>
<li><a href="oefinder-analysis-ercc.html">ERCC genes</a></li>
<li><a href="oefinder-analysis-cpm-batch.html">Endogeneous genes per batch, per individual</a></li>
</ul></li>
<li><a href="gene-symbol.html">Extract gene symbols</a></li>
<li>Cell-to-cell variation analysis
<ul>
<li><a href="cv-sparsity.html">CV and sparsity</a></li>
<li>Mean-adjusted CV – ( <a href="cv-adjusted-wo-19098-r2.html">Normalize CV excluding outlier batch</a>, <a href="cv-adjusted.html">Normalize CV including outlier batch</a>, <a href="cv-adjusted-profile.html">Gene rankings</a>, <a href="cv-adjusted-profile-annotation.html">Annotations</a>, <a href="cv-adjusted-comparison.html">Gene-level dissimilarity measure</a> )</li>
<li>Compare mean-adjusted CVs – (<a href="cv-adjusted-all-genes-wo-19098-2.html">ANOVA</a>, <a href="cv-annotation.html">ANOVA &amp; annotations</a>, <a href="cv-adjusted-statistical-test.html">Sum of Squared Deviation from the Median</a>, <a href="cv-adjusted-statistical-test-permute.html">Permutation-based p-values</a> )</li>
<li>Noisy genes analysis – (<a href="cv-adjusted-summary.html">final and Gaussian-based transformation</a>, <a href="cv-adjusted-summary-pois.html">final and Poisson-based transformation</a>, <a href="compare-distribution-pluripotency.html">Pluripotent gene density plots</a> )</li>
<li>Putting it all together… (<a href="cv-adjusted-summary-pois.html">All cells</a>, <a href="cv-adjusted-summary-pois-expressed.html">Expressed cells</a>, <a href="cv-adjusted-summary-pois-dropout.html">Undetected cells</a> )</li>
</ul></li>
<li><a href="qc-plots.html">Quality control plots</a></li>
<li><a href="total-molecule.html">Total molecule-counts and standardization</a></li>
<li>Exploring sequence coverage
<ul>
<li>Endogenous genes
<ul>
<li><a href="coverage-endogenous.html">molecules</a></li>
<li><a href="coverage-endogenous-reads.html">reads</a></li>
<li><a href="coverage-endogenous-bulk.html">bulk</a></li>
</ul></li>
<li>ERCC spike-ins
<ul>
<li><a href="coverage-ercc.html">molecules</a></li>
<li><a href="coverage-ercc-reads.html">reads</a></li>
<li><a href="coverage-ercc-bulk.html">bulk</a></li>
</ul></li>
<li>Using ngsplot
<ul>
<li><a href="ngsplot-setup.html">Setup</a></li>
<li><a href="ngsplot-endogenous.html">endogenous genes</a></li>
<li><a href="ngsplot-ercc.html">ERCC genes</a></li>
</ul></li>
</ul></li>
<li><a href="pca-quantification.html">PCA quantification</a></li>
</ul>
</div>
<div id="download-data" class="section level2">
<h2>Download data</h2>
<ul>
<li><a href="download_flow_cell_01.html">Download flow cell 01 - C6WYKACXX</a></li>
<li><a href="download_flow_cell_02.html">Download flow cell 02 - C6WURACXX</a></li>
<li><a href="download_flow_cell_03.html">Download flow cell 03 - C723YACXX</a></li>
<li><a href="download_flow_cell_04.html">Download flow cell 04 - C72JMACXX</a></li>
</ul>
</div>
<div id="one-time-investigations" class="section level2">
<h2>One-time investigations</h2>
<ul>
<li><a href="verify-md5sum.html">Verify md5 checksums</a></li>
<li><a href="files-for-testing.html">Small and large files for testing</a></li>
<li><a href="bug-conversion-01.html">Tracking down bug in reads to molecules conversion - Attempt 01</a></li>
<li><a href="read2pos.html">The effect of using featureCounts read2pos flag</a></li>
<li><a href="bug-conversion-02.html">Tracking down bug in reads to molecules conversion - Attempt 02</a></li>
<li><a href="bug-conversion-03.html">Tracking down bug in reads to molecules conversion - Attempt 03</a></li>
<li><a href="bug-conversion-04.html">Tracking down bug in reads to molecules conversion - Attempt 04</a></li>
</ul>
</div>
<div id="other" class="section level2">
<h2>Other</h2>
<ul>
<li><a href="analysis.html">Assessment of rapid run flow cell</a></li>
</ul>
</div>
<div id="lcl-data-from-a-full-flowcell" class="section level2">
<h2>LCL data from a full flowcell</h2>
<ul>
<li><a href="qc-cell-library.html">Quality control of cell/library</a></li>
</ul>
<p><br> <img src="divvy.jpg" alt="Divvy bikes" /> <br> <em><a href="https://www.flickr.com/photos/jamesbondsv/9041673199">Photo</a> by <a href="https://www.flickr.com/photos/jamesbondsv">Steven Vance</a> / <a href="https://creativecommons.org/licenses/by/2.0/">CC BY 2.0</a></em></p>
</div>
<div id="project-overview" class="section level2">
<h2>Project overview</h2>
<p>The purpose of this project is to gain some insight into city-wide biking trends by analyzing the <a href="https://www.divvybikes.com/system-data">Divvy trip data</a>. Also, I examine trip data from one bike station at the University of Chicago to compare the biking patterns at the university against city-wide trends.</p>
<p>All the results and plots presented in the pages below should be reproduceable on your computer. Follow the <a href="setup.html">Setup Instructions</a> if you are interested in reproducing the results for yourself.</p>
<p>These are the results of my analyses. They were generated by rendering the <a href="https://github.com/stephenslab/wflow-divvy/tree/master/analysis">R Markdown documents</a> into webpages.</p>
<ol style="list-style-type: decimal">
<li><p><a href="first-glance.html">A first glance at the Divvy data.</a></p></li>
<li><p><a href="station-map.html">A map of the Divvy stations in Chicago.</a></p></li>
<li><p><a href="time-of-day-trends.html">Exploring daily bike commuting trends from the Divvy data.</a></p></li>
<li><p><a href="seasonal-trends.html">Exploring seasonal biking trends from the Divvy data.</a></p></li>
</ol>
</div>
<div id="credits" class="section level2">
<h2>Credits</h2>
<p>This workflowr project was developed by <a href="http://pcarbo.github.io">Peter Carbonetto</a> at the <a href="https://www.uchicago.edu">University of Chicago</a>.</p>
<p>Thanks to <a href="https://github.com/jdblischak">John Blischak</a> and <a href="http://stephenslab.uchicago.edu">Matthew Stephens</a> for their assistance and support. Also, thanks to <a href="https://rstudio-pubs-static.s3.amazonaws.com/63061_90f5136ffdf74740b6ba4ad8f2fd72fe.html">Larry Layne</a> and <a href="http://www.austinwehrwein.com/data-visualization/heatmaps-with-divvy-data">Austin Wehrwein</a> for sharing their analyses of the Divvy trip data that inspired some of the investigations here.</p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.5.3 (2019-03-11)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

Matrix products: default
BLAS: /data/sgg2/jenny/bin/R-3.5.3/lib64/R/lib/libRblas.so
LAPACK: /data/sgg2/jenny/bin/R-3.5.3/lib64/R/lib/libRlapack.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

loaded via a namespace (and not attached):
 [1] workflowr_1.5.0 Rcpp_1.0.3      rprojroot_1.3-2 digest_0.6.23  
 [5] later_1.0.0     R6_2.4.1        backports_1.1.5 git2r_0.26.1   
 [9] magrittr_1.5    evaluate_0.14   stringi_1.4.3   rlang_0.4.1    
[13] fs_1.3.1        promises_1.1.0  whisker_0.4     rmarkdown_1.18 
[17] tools_3.5.3     stringr_1.4.0   glue_1.3.1      httpuv_1.5.2   
[21] xfun_0.11       yaml_2.2.0      compiler_3.5.3  htmltools_0.4.0
[25] knitr_1.26     </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>



</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
