<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Jenny Sjaarda" />


<title>Data sources</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/readable.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/kePrint-0.0.1/kePrint.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">PSYMETAB</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="setup.html">Setup</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/jennysjaarda/PSYMETAB">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Data sources</h1>
<h4 class="author">Jenny Sjaarda</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2020-06-12
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>PSYMETAB/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20191126code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20191126)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20191126code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20191126)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomjennysjaardaPSYMETABtreeff40613a707596f82f8e567484f57b7ceab5842atargetblankff40613a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/jennysjaarda/PSYMETAB/tree/ff40613a707596f82f8e567484f57b7ceab5842a" target="_blank">ff40613</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomjennysjaardaPSYMETABtreeff40613a707596f82f8e567484f57b7ceab5842atargetblankff40613a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    ._docs
    Ignored:    .drake/
    Ignored:    analysis/.Rhistory
    Ignored:    analysis/._GWAS.Rmd
    Ignored:    analysis/._data_processing_in_genomestudio.Rmd
    Ignored:    analysis/._quality_control.Rmd
    Ignored:    analysis/GWAS/
    Ignored:    analysis/PRS/
    Ignored:    analysis/QC/
    Ignored:    analysis/figure/
    Ignored:    analysis_prep_1_clustermq.out
    Ignored:    analysis_prep_2_clustermq.out
    Ignored:    analysis_prep_3_clustermq.out
    Ignored:    analysis_prep_4_clustermq.out
    Ignored:    data/processed/
    Ignored:    data/raw/
    Ignored:    download_impute_1_clustermq.out
    Ignored:    init_analysis_1_clustermq.out
    Ignored:    init_analysis_2_clustermq.out
    Ignored:    init_analysis_3_clustermq.out
    Ignored:    init_analysis_4_clustermq.out
    Ignored:    init_analysis_5_clustermq.out
    Ignored:    init_analysis_6_clustermq.out
    Ignored:    packrat/lib-R/
    Ignored:    packrat/lib-ext/
    Ignored:    packrat/lib/
    Ignored:    post_impute_1_clustermq.out
    Ignored:    pre_impute_qc_1_clustermq.out
    Ignored:    process_init_10_clustermq.out
    Ignored:    process_init_11_clustermq.out
    Ignored:    process_init_12_clustermq.out
    Ignored:    process_init_13_clustermq.out
    Ignored:    process_init_14_clustermq.out
    Ignored:    process_init_15_clustermq.out
    Ignored:    process_init_16_clustermq.out
    Ignored:    process_init_17_clustermq.out
    Ignored:    process_init_18_clustermq.out
    Ignored:    process_init_19_clustermq.out
    Ignored:    process_init_1_clustermq.out
    Ignored:    process_init_20_clustermq.out
    Ignored:    process_init_21_clustermq.out
    Ignored:    process_init_22_clustermq.out
    Ignored:    process_init_23_clustermq.out
    Ignored:    process_init_24_clustermq.out
    Ignored:    process_init_25_clustermq.out
    Ignored:    process_init_26_clustermq.out
    Ignored:    process_init_27_clustermq.out
    Ignored:    process_init_28_clustermq.out
    Ignored:    process_init_29_clustermq.out
    Ignored:    process_init_2_clustermq.out
    Ignored:    process_init_30_clustermq.out
    Ignored:    process_init_31_clustermq.out
    Ignored:    process_init_3_clustermq.out
    Ignored:    process_init_4_clustermq.out
    Ignored:    process_init_5_clustermq.out
    Ignored:    process_init_6_clustermq.out
    Ignored:    process_init_7_clustermq.out
    Ignored:    process_init_8_clustermq.out
    Ignored:    process_init_9_clustermq.out
    Ignored:    prs_1_clustermq.out
    Ignored:    prs_2_clustermq.out
    Ignored:    prs_3_clustermq.out
    Ignored:    prs_4_clustermq.out

Untracked files:
    Untracked:  analysis/genetic_quality_control.Rmd
    Untracked:  analysis/pheno_quality_control.Rmd
    Untracked:  analysis/plans.Rmd
    Untracked:  analysis_prep.log
    Untracked:  download_impute.log
    Untracked:  file1.txt
    Untracked:  file2.txt
    Untracked:  file3.txt
    Untracked:  flagged_rows.txt
    Untracked:  grs.log
    Untracked:  init_analysis.log
    Untracked:  problem_drugs.txt
    Untracked:  problem_ids.txt
    Untracked:  process_init.log
    Untracked:  prs.log

Unstaged changes:
    Modified:   analysis/GWAS.Rmd
    Modified:   analysis/data_sources.Rmd
    Deleted:    analysis/project.Rmd
    Modified:   analysis/quality_control.Rmd
    Modified:   cache_log.csv
    Modified:   post_impute.log
    Modified:   slurm_clustermq.tmpl

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the R Markdown and HTML files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view them.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jennysjaarda/PSYMETAB/blob/e0d60323954813d80b15ff6d9d76c6f2c9187eda/analysis/data_sources.Rmd" target="_blank">e0d6032</a>
</td>
<td>
Sjaarda Jennifer Lynn
</td>
<td>
2019-12-06
</td>
<td>
edit typos
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jennysjaarda/PSYMETAB/46477dd771ba63ffa5d44a0d4f53a1a835a71c87/docs/data_sources.html" target="_blank">46477dd</a>
</td>
<td>
Jenny Sjaarda
</td>
<td>
2019-12-06
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jennysjaarda/PSYMETAB/blob/b503ef05e45518dbd64f6b84c3c6f4e0d0fd64f8/analysis/data_sources.Rmd" target="_blank">b503ef0</a>
</td>
<td>
Sjaarda Jennifer Lynn
</td>
<td>
2019-12-06
</td>
<td>
add more details to website
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jennysjaarda/PSYMETAB/9f1ba5e2539136e5a4cdcbba996ba481e11d4cc0/docs/data_sources.html" target="_blank">9f1ba5e</a>
</td>
<td>
Jenny Sjaarda
</td>
<td>
2019-12-06
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jennysjaarda/PSYMETAB/blob/c1d579b9f7e71e91086b8652c807db5bcb796cee/analysis/data_sources.Rmd" target="_blank">c1d579b</a>
</td>
<td>
Jenny
</td>
<td>
2019-12-04
</td>
<td>
add details on data
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p><em>provided from iGE3</em></p>
<div id="general-information" class="section level2">
<h2>General information</h2>
<ul>
<li>Files received from: Mylene Docquier (<a href="mailto:Mylene.Docquier@unige.ch">Mylene.Docquier@unige.ch</a>), iGE3 Genomics Platform Manager, University of Geneva</li>
<li>ftp server details:
<ul>
<li>host: <a href="sftp://129.194.88.17" class="uri">sftp://129.194.88.17</a></li>
<li>username: 080219CE</li>
<li>password: chinjennys</li>
</ul></li>
<li>Genotype data received in genomestudio format on August 28, 2019; for processing and converting to PLINK format see docs/miscellaneous/data_processing.md</li>
</ul>
</div>
<div id="genotype-data" class="section level2">
<h2>Genotype data</h2>
<ul>
<li>Genotype data found in data/raw/genotypes</li>
<li>Each folder contains:
<ul>
<li>Initial data provided from Mylene in genomestudio format, with the original folder name (‘XXX/’).</li>
<li>Cluster files in genomestudio format (see docs/miscellaneous/data_processing.md), and named ‘XXX_cluster/’.</li>
<li>PLINK files exported from genomestudio.</li>
</ul></li>
</ul>
</div>
<div id="miscellaneous-gsa-information-provided-in-the-following-files" class="section level2">
<h2>Miscellaneous GSA information provided in the following files:</h2>
<ol style="list-style-type: decimal">
<li>GSA v2 + MD Consortium.csv</li>
<li>GSAMD-24v2-0_20024620_A1.csv</li>
<li>GSAMD-24v2-0_A1-ACMG-GeneAnnotation.xlsx</li>
<li>GSAMD-24v2-0_A1-ADME-CPIC-GeneAnnotation.xlsx</li>
<li>GSAMD-24v2-0_A1-HLA-GeneAnnotation.xlsx</li>
<li>GSAMD-24v2-0_A1-TruSight-GeneAnnotation.xlsx</li>
<li>GSAv2_MDConsortium.bpm</li>
<li>GSPMA24v1_0-A_4349HNR_Samples.egt</li>
</ol>
<div id="details" class="section level3">
<h3>Details</h3>
<ul>
<li>Files 1 and 2 appear to be identical and correspond to strand illumina strand information, same file can be found <a href="http://glimdna.org/assets/gsamd-24v2-0_20024620_a1.zip">here</a>.</li>
<li>xlsx files contain 2 tabs: “Coverage Summary” and “GSAMD-24v2-0_A1-XXX-GeneAnnota”</li>
<li>bpm file corresponds to manifest file for use in genomestudio. Manifest files provide a description of the SNP or probe content on a standard BeadChip or in an assay product.</li>
<li>egt file corresponds to cluster file for making genotype calls.</li>
<li>all saved in data/reference_files</li>
</ul>
</div>
</div>
<div id="chip-details-from-illumina" class="section level2">
<h2>Chip details from Illumina</h2>
<ul>
<li>Files received from: Fe Magbanua (<a href="mailto:techsupport@illumina.com">techsupport@illumina.com</a>),Technical Applications Scientist, Technical Support, Illumina</li>
<li>GSAMD-24v2-0_20024620_A4_StrandReport_FDT.txt: strand report build38 (build37 not available).</li>
<li>GSAMD-24v2-0_20024620_A1_b151_rsids.txt: loci to rsid conversion file build37.</li>
<li>GSAMD-24v2-0_20024620_A4_b151_rsids.txt: loci to rsid conversion file build38.</li>
<li>all saved in data/reference_files (copied using FileZilla)</li>
</ul>
</div>
<div id="strand-files-from-welcome-centre" class="section level2">
<h2>Strand files from Welcome Centre</h2>
<ul>
<li>The data for each chip and genome build combination are freely downloadable from the links localted <a href="https://www.well.ox.ac.uk/~wrayner/strand/">here</a>, each zip file contains three files, these are:
<ul>
<li><code>.strand</code> file</li>
<li><code>.miss</code> file</li>
<li><code>.multiple</code> file</li>
</ul></li>
<li>More details can be found at the link above</li>
<li><a href="http://mccarthy.well.ox.ac.uk/chipendium/ui/">Chipendium</a> was used to comfirm that bim files are on the TOP strand .</li>
<li>Contacted William Rayner (<a href="mailto:wrayner@well.ox.ac.uk">wrayner@well.ox.ac.uk</a>) to find out what to do about custom SNPs, all correspondence on 22/07/2019.
<ul>
<li>Query:</li>
</ul>
<blockquote>
<p>The chip used to generate the data was the GSAMD-24v2, however about 10,000 custom SNPs were also added to the chip. Do you have any recommendations for adding such SNPs to the strand file for processing?</p>
</blockquote>
<ul>
<li>Response:</li>
</ul>
<blockquote>
<p>If you have a chip with custom content on it as you do if you are able to send me the .csv annotation file (that contains the TopGenomicSeq information) I can use that to create you a custom strand file that you can then download on a private link, this will ensure the extra SNPs are not lost in the strand update (at the moment they would be removed as non-matching).</p>
</blockquote>
<ul>
<li>Trying to obtain such .csv file from Mylene or Smita at Illumina (<a href="mailto:spathak@illumina.com">spathak@illumina.com</a>) who designed the chip.</li>
<li>On 15/07/2019 Smita provided such a file: GSA_UPPC_20023490X357589_A1_custom_only.csv.</li>
<li>The file was downloaded and save to UPPC (Jenny/PSYMETAB_GWAS/GSA).</li>
<li>Sent .csv file to William Rayner and he provided the strand file for the custom SNP list on 16/07/2019:
<ul>
<li>GSA_UPPC_20023490X357589_A1_custom_only-b37-strand.zip</li>
<li>GSA_UPPC_20023490X357589_A1_custom_only-b37-strand.zip</li>
</ul></li>
<li>Zipped strand files were copied to SGG server (${project_dir}/data/raw/reference_files/) and subsequently unzipped and used in QC (only b37 files was needed).</li>
</ul></li>
</ul>
</div>
<div id="phenotype-data" class="section level2">
<h2>Phenotype data</h2>
<div id="sexethnicity-data" class="section level3">
<h3>Sex/ethnicity data</h3>
<ul>
<li>Provided by Celine (via email) for each batch on July 18, 2019: <code>GSA_sex-ethnicity.xlsx</code>.</li>
<li>Used for genetic quality control.</li>
<li>Downloaded and saved to UPPC folder (Jenny/PSYMETAB_GWAS/).</li>
<li>Opened, manually changed all accents to standard letters (ctrl-F and replace) and re-saved as csv/xlsx file (with <code>no_accents</code>) for easier use in R.</li>
<li>Moved to SGG folders via filezilla (manually).</li>
<li>Name was changed, as follows:</li>
</ul>
<pre class="bash"><code>mv data/raw/phenotype_data/GSA_sex-ethnicity.xlsx data/raw/phenotype_data/QC_sex_eth.xlsx</code></pre>
</div>
<div id="full-phenotype-data" class="section level3">
<h3>Full phenotype data</h3>
<ul>
<li>Phenotype data to be used in GWAS (and other) analyses.</li>
<li>All data was saved on sgg folders to: <code>PSYMETAB/data/raw/phenotype_data/</code>.</li>
<li>Iniitial data provided by Celine (via email) on October 24, 2019: <code>PHENO_GWAS_241019.xlsx</code>.
<ul>
<li>Saved to sgg folder <code>PSYMETAB/data/phenotype_data/</code>.</li>
<li>Opened in excel, saved as a <code>.csv</code> file.</li>
<li>Manually changed all accents to standard letters (ctrl-F and replace) and re-saved as csv file (with extension <code>_no_accents.csv</code>) for easier use in R.</li>
</ul></li>
<li>Discovered problems with phenotype database in early 2020, new phenotype data was provided on March 16, 2020: <code>PHENO_GWAS_160320.xlsx</code>.
<ul>
<li>Same process as above was followed to remove accents and convert to csv file.</li>
<li>Details on problems detected can be found on the <a href="pheno_quality_control.html">Phenotype Quality Control page</a>.</li>
</ul></li>
<li>Additional problems were discovered and a new dataset was provided on April 16, 2020: <code>PHENO_GWAS_160420.xlsx</code> (processed the same as above).</li>
</ul>
</div>
<div id="caffeine-data" class="section level3">
<h3>Caffeine data</h3>
<ul>
<li>Requested to perform analysis using Nermine’s caffeine data.</li>
<li>All data was saved on sgg folders to: <code>PSYMETAB/data/raw/phenotype_data/</code>.</li>
<li>Initial data provided by Claire (via email) on March 11, 2020: <code>Code_GEN_CG_11.03.2020.xlsx</code>.</li>
<li>New variables (data of blood draw and age) were provided by Claire on May 5, 2020: <code>CAF_Sleep_Jenny_05_05_2020_CG.xlsx</code>.
<ul>
<li>Noticed that there was a problem with the data set (email describing problem to Nermine on 07/05/2020):</li>
</ul>
<blockquote>
<p>I am a bit confused by the age and date columns in some cases. For example, for the participant below we have measurements at 4 time points: age 34, 52, 53 and 39 and the date of measurement in the next column. Does the date correspond to the date the blood draw took place? If so, I don’t understand why in 2018 (row 4) this participant was 39 and in 2012 (row 3) they were 53! Or does this column represent the date caffeine was measured (which could be very different from date of extraction)? Ideally I need the age the participant was when the extraction took place.</p>
</blockquote>
<table class="table" style="width: auto !important; ">
<thead>
<tr>
<th style="text-align:left;">
GEN
</th>
<th style="text-align:right;">
age
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
UXEWHQEZ
</td>
<td style="text-align:right;">
52
</td>
<td style="text-align:left;">
2011-04-26
</td>
</tr>
<tr>
<td style="text-align:left;">
UXEWHQEZ
</td>
<td style="text-align:right;">
34
</td>
<td style="text-align:left;">
2013-04-02
</td>
</tr>
<tr>
<td style="text-align:left;">
UXEWHQEZ
</td>
<td style="text-align:right;">
52
</td>
<td style="text-align:left;">
2011-03-24
</td>
</tr>
<tr>
<td style="text-align:left;">
UXEWHQEZ
</td>
<td style="text-align:right;">
53
</td>
<td style="text-align:left;">
2012-02-27
</td>
</tr>
<tr>
<td style="text-align:left;">
UXEWHQEZ
</td>
<td style="text-align:right;">
39
</td>
<td style="text-align:left;">
2018-05-07
</td>
</tr>
</tbody>
</table>
<ul>
<li>After discussing with Nermine and Claire, Claire discovered the problem (as per email on 07/05/2020):</li>
</ul>
<blockquote>
<p>I spotted the problem, it was a mismatch in the ambulatory codes as some of them are written XXXAMB instead of AMBXXX or XXX+letter. I will send Jenny a new version of the file with paying attention to these codes if ok for you?</p>
</blockquote></li>
<li>A New dataset was provided by Cliare on May 7, 2020: <code>CAF_Sleep_Jenny_07_05_2020_CG.xlsx</code>.
<ul>
<li>There were still issues with this dataset (as per email on 07/07/2020):</li>
</ul>
<blockquote>
<p>Il y a toujours un petit problème avec les IDs : GAWZCNNL et WIFRYNSK. Je mets les problèmes en dessous. Je crois que deux participants ont reçu le meme GEN ID. Peut-être il y avait encore un problème avec le merge ?</p>
</blockquote>
<div style="border: 1px solid #ddd; padding: 5px; overflow-x: scroll; width:100%; ">
<table class="table" style="width: auto !important; ">
<thead>
<tr>
<th style="text-align:left;">
GEN
</th>
<th style="text-align:left;">
Date
</th>
<th style="text-align:right;">
age
</th>
<th style="text-align:right;">
age2
</th>
<th style="text-align:left;">
Date2
</th>
<th style="text-align:right;">
days_difference
</th>
<th style="text-align:right;">
age_plus_days
</th>
<th style="text-align:left;">
age_check
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
GAWZCNNL
</td>
<td style="text-align:left;">
2011-02-03
</td>
<td style="text-align:right;">
53
</td>
<td style="text-align:right;">
22
</td>
<td style="text-align:left;">
2018-02-22
</td>
<td style="text-align:right;">
2576
</td>
<td style="text-align:right;">
60.05753
</td>
<td style="text-align:left;">
problem
</td>
</tr>
<tr>
<td style="text-align:left;">
GAWZCNNL
</td>
<td style="text-align:left;">
2011-02-03
</td>
<td style="text-align:right;">
53
</td>
<td style="text-align:right;">
60
</td>
<td style="text-align:left;">
2018-05-08
</td>
<td style="text-align:right;">
2651
</td>
<td style="text-align:right;">
60.26301
</td>
<td style="text-align:left;">
sensible
</td>
</tr>
<tr>
<td style="text-align:left;">
GAWZCNNL
</td>
<td style="text-align:left;">
2018-02-22
</td>
<td style="text-align:right;">
22
</td>
<td style="text-align:right;">
60
</td>
<td style="text-align:left;">
2018-05-08
</td>
<td style="text-align:right;">
75
</td>
<td style="text-align:right;">
22.20548
</td>
<td style="text-align:left;">
sensible
</td>
</tr>
<tr>
<td style="text-align:left;">
WIFRYNSK
</td>
<td style="text-align:left;">
2009-01-19
</td>
<td style="text-align:right;">
31
</td>
<td style="text-align:right;">
31
</td>
<td style="text-align:left;">
2018-09-10
</td>
<td style="text-align:right;">
3521
</td>
<td style="text-align:right;">
40.64658
</td>
<td style="text-align:left;">
problem
</td>
</tr>
</tbody>
</table>
</div></li>
<li>Celine performed the anonymization to compare to Claire’s data to see if the same problems arose: <code>GEN_CAF_Sleep_Jenny_05_05_2020.xlsx</code>.
<ul>
<li>There was still one issue with the dataset at ID WIFRYNSK.
<table class="table" style="width: auto !important; ">
<thead>
<tr>
<th style="text-align:left;">
GEN
</th>
<th style="text-align:left;">
Date
</th>
<th style="text-align:right;">
age
</th>
<th style="text-align:right;">
age2
</th>
<th style="text-align:left;">
Date2
</th>
<th style="text-align:right;">
days_difference
</th>
<th style="text-align:right;">
age_plus_days
</th>
<th style="text-align:left;">
age_check
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
WIFRYNSK
</td>
<td style="text-align:left;">
2009-01-07
</td>
<td style="text-align:right;">
31
</td>
<td style="text-align:right;">
31
</td>
<td style="text-align:left;">
2018-08-29
</td>
<td style="text-align:right;">
3521
</td>
<td style="text-align:right;">
40.64658
</td>
<td style="text-align:left;">
problem
</td>
</tr>
</tbody>
</table></li>
<li>After asking Celine about this problem (08/05/2020), she said:</li>
</ul>
<blockquote>
<p>Okay alors j’ai vérifié et cette erreur est déjà présente dans le fichier que Nermine nous a envoyé, donc ce n’est pas un problème lié au changement de codes.</p>
</blockquote>
<ul>
<li>Nermine responded (08/05/2020) with the explanation:</li>
</ul>
<blockquote>
<p>Je viens de vérifier et en effet j’avais changé l’âge manuellement et il était changé dans les deux observations même si j’ai précisé la date de l’observation… je ne sais pas comment j’ai fais. Alors à la date du 22.12.2008, le patient avait 21 ans, et à la date du 13.08.2018 le patient avait 31 ans.</p>
</blockquote>
<ul>
<li>The following line was added to the script to manually change this entry: <code>caffeine_raw %&gt;% mutate(age=replace(age, GEN==&quot;WIFRYNSK&quot; &amp; as.Date(Date)==&quot;2009-01-07&quot;, 21)</code></li>
</ul></li>
</ul>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.5.3 (2019-03-11)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

Matrix products: default
BLAS: /data/sgg2/jenny/bin/R-3.5.3/lib64/R/lib/libRblas.so
LAPACK: /data/sgg2/jenny/bin/R-3.5.3/lib64/R/lib/libRlapack.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] fuzzyjoin_0.1.5         kableExtra_1.1.0        R.utils_2.9.2          
 [4] R.oo_1.23.0             R.methodsS3_1.7.1       TwoSampleMR_0.4.25     
 [7] reader_1.0.6            NCmisc_1.1.6            optparse_1.6.4         
[10] readxl_1.3.1            ggthemes_4.2.0          tryCatchLog_1.1.6      
[13] futile.logger_1.4.3     DataExplorer_0.8.0      taRifx_1.0.6.1         
[16] qqman_0.1.4             MASS_7.3-51.5           bit64_0.9-7            
[19] bit_1.1-14              rslurm_0.5.0            rmeta_3.0              
[22] devtools_2.2.1          usethis_1.5.1           data.table_1.12.8      
[25] clustermq_0.8.8.1       future.batchtools_0.8.1 future_1.15.1          
[28] rlang_0.4.5             knitr_1.26              drake_7.12.0.9000      
[31] forcats_0.4.0           stringr_1.4.0           dplyr_0.8.3            
[34] purrr_0.3.3             readr_1.3.1             tidyr_1.0.3            
[37] tibble_2.1.3            ggplot2_3.2.1           tidyverse_1.3.0        
[40] pacman_0.5.1            processx_3.4.1          workflowr_1.6.0        

loaded via a namespace (and not attached):
 [1] backports_1.1.6      plyr_1.8.5           igraph_1.2.5        
 [4] lazyeval_0.2.2       storr_1.2.1          listenv_0.8.0       
 [7] digest_0.6.25        htmltools_0.4.0      fansi_0.4.1         
[10] magrittr_1.5         checkmate_1.9.4      memoise_1.1.0       
[13] base64url_1.4        remotes_2.1.0        globals_0.12.5      
[16] modelr_0.1.5         prettyunits_1.1.0    colorspace_1.4-1    
[19] rvest_0.3.5          rappdirs_0.3.1       haven_2.2.0         
[22] xfun_0.11            callr_3.4.0          crayon_1.3.4        
[25] jsonlite_1.6         brew_1.0-6           glue_1.4.0          
[28] gtable_0.3.0         webshot_0.5.2        pkgbuild_1.0.6      
[31] scales_1.1.0         futile.options_1.0.1 DBI_1.1.0           
[34] Rcpp_1.0.3           viridisLite_0.3.0    progress_1.2.2      
[37] txtq_0.2.0           htmlwidgets_1.5.1    httr_1.4.1          
[40] getopt_1.20.3        calibrate_1.7.5      ellipsis_0.3.0      
[43] pkgconfig_2.0.3      dbplyr_1.4.2         tidyselect_0.2.5    
[46] reshape2_1.4.3       later_1.0.0          munsell_0.5.0       
[49] cellranger_1.1.0     tools_3.5.3          cli_2.0.1           
[52] generics_0.0.2       broom_0.5.3          evaluate_0.14       
[55] yaml_2.2.0           fs_1.3.1             packrat_0.5.0       
[58] nlme_3.1-143         whisker_0.4          formatR_1.7         
[61] proftools_0.99-2     xml2_1.2.2           compiler_3.5.3      
[64] rstudioapi_0.10      filelock_1.0.2       testthat_2.3.1      
[67] reprex_0.3.0         stringi_1.4.5        highr_0.8           
[70] ps_1.3.0             desc_1.2.0           lattice_0.20-38     
[73] vctrs_0.2.4          pillar_1.4.3         lifecycle_0.1.0     
[76] networkD3_0.4        httpuv_1.5.2         R6_2.4.1            
[79] promises_1.1.0       gridExtra_2.3        sessioninfo_1.1.1   
[82] codetools_0.2-16     lambda.r_1.2.4       assertthat_0.2.1    
[85] pkgload_1.0.2        rprojroot_1.3-2      withr_2.1.2         
[88] batchtools_0.9.12    parallel_3.5.3       hms_0.5.3           
[91] grid_3.5.3           rmarkdown_1.18       git2r_0.26.1        
[94] lubridate_1.7.4     </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
