<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Jenny Sjaarda" />


<title>Data Processing in GenomeStudio</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/readable.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->



<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">PSYMETAB</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="setup.html">Setup</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/jennysjaarda/PSYMETAB">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Data Processing in GenomeStudio</h1>
<h4 class="author">Jenny Sjaarda</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2019-12-02
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 2 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>PSYMETAB/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.5.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomjennysjaardaPSYMETABtree0dd02a792ccdb998d5650da4f6b713f0d4a46c83targetblank0dd02a7a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/jennysjaarda/PSYMETAB/tree/0dd02a792ccdb998d5650da4f6b713f0d4a46c83" target="_blank">0dd02a7</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomjennysjaardaPSYMETABtree0dd02a792ccdb998d5650da4f6b713f0d4a46c83targetblank0dd02a7a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .drake/
    Ignored:    data/processed/
    Ignored:    data/raw/

Untracked files:
    Untracked:  analysis/QC/
    Untracked:  post_imputation_qc.log
    Untracked:  pre_impute_qc.out
    Untracked:  qc_part2.out

Unstaged changes:
    Deleted:    pre_imputation_qc.out
    Deleted:    qc_part1.out

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the R Markdown and HTML files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view them.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jennysjaarda/PSYMETAB/blob/0dd02a792ccdb998d5650da4f6b713f0d4a46c83/analysis/data_processing_in_genomestudio.Rmd" target="_blank">0dd02a7</a>
</td>
<td>
Jenny
</td>
<td>
2019-12-02
</td>
<td>
modify website
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="creating-genomestudio-files" class="section level2">
<h2>Creating GenomeStudio files:</h2>
<ul>
<li>Instructions can be found <a href="https://support.illumina.com/content/dam/illumina-support/documents/documentation/software_documentation/genomestudio/genomestudio-2-0/genomestudio-genotyping-module-v2-user-guide-11319113-01.pdf">here</a>.</li>
<li>Required files:
<ul>
<li>Sample sheet: as csv file.</li>
<li>Data repository: as idat files.</li>
<li>Manifest file: as bpm file.</li>
<li>Cluster file: as egt file.</li>
</ul></li>
<li>Data provided from Mylene Docquier, copied from sftp and saved here: <code>L:\PCN\UBPC\ANALYSES_RECHERCHE\Jenny\PSYMETAB_GWAS\data</code>.</li>
<li>Create new IDs based on GPCR randomization (see /scripts/format_QC_input.r), and save to above folder as: <code>Eap0819_1t26_27to29corrected_7b9b_randomizedID.csv</code>.</li>
<li>Note that original IDs can be found in the same folder at the file: <code>Eap0819_1t26_27to29corrected_7b9.csv</code>, if needed.</li>
<li>Create empty folder here: <code>L:\PCN\UBPC\ANALYSES_RECHERCHE\Jenny\PSYMETAB_GWAS</code>, named: <code>GS_project_26092019</code> (data of creation).</li>
<li>Using new IDs, create genome studio project as follows:
<ol style="list-style-type: decimal">
<li>Open GenomeStudio.</li>
<li>Select: File &gt; New Genotyping Project.</li>
<li>Select <code>L:\PCN\UBPC\ANALYSES_RECHERCHE\Jenny\PSYMETAB_GWAS</code> as project repository.</li>
<li>Under project name: use “GS_project_26092019” and click “Next”.</li>
<li>Select “Use sample sheet to load intensities” and click “Next”.</li>
<li>Select sample, data and manifests as specified below and click “Next”:
<ul>
<li>Sample sheet: <code>L:\PCN\UBPC\ANALYSES_RECHERCHE\Jenny\PSYMETAB_GWAS\data\Eap0819_1t26_27to29corrected_7b9b_randomizedID.csv</code>,</li>
<li>Data repository: <code>L:\PCN\UBPC\ANALYSES_RECHERCHE\Jenny\PSYMETAB_GWAS\data</code>,</li>
<li>Manifest repository: <code>L:\PCN\UBPC\ANALYSES_RECHERCHE\Jenny\PSYMETAB_GWAS\data</code>.</li>
</ul></li>
<li>Select “Import cluster positions from cluster file” and choose cluster file located here: <code>L:\PCN\UBPC\ANALYSES_RECHERCHE\Jenny\PSYMETAB_GWAS\data\GSPMA24v1_0-A_4349HNR_Samples.egt</code> and click “Finish”.</li>
</ol></li>
</ul>
</div>
<div id="clustering-and-plink-conversion" class="section level2">
<h2>Clustering and PLINK conversion</h2>
<ul>
<li>Data saved on CHUV servers at the following location: <code>L:\PCN\UBPC\ANALYSES_RECHERCHE\Jenny\PSYMETAB_GWAS\GS_project_26092019</code>.</li>
<li><code>GS_project_26092019.bsc</code> was opened (requires Genome Studio) and used for clustering.</li>
<li>Clustering was performed according to the guidelines in the webinar (see <a href="creating_custom_cluster_files.html">Notes on creating custom cluster files</a>) with the following procedure:
<ol style="list-style-type: decimal">
<li>Cluster: cluster all SNPs, evaluate samples.
<ol style="list-style-type: decimal">
<li>Once data was opened, all SNPs were clustered by clicking “Cluster all SNPs button” in the top panel (icon with 3 red, purple, blue ovals). SNP statistics and heritability estimate updates were ignored.</li>
<li>In the samples table (bottom left panel) call rate was recalculated by clicking “Calculate” (calculator icon) to recalculate SNP call rates with new cluster positions. SNP statistics and heritability estimate updates were ignored.</li>
<li>The sample table was then sorted by the “Call Rate” column and samples with call rate &lt;95% were selected and removed from downstream processing (right click, select “Exclude Selected Samples”), 8 samples fell below this cut-off. Updates were ignored. In “SNP Graph” right click and deselect “Show Excluded Samples”.</li>
</ol></li>
<li>Recluster: cluster sex chromosomes, cluster autosomes.
<ol style="list-style-type: decimal">
<li>“SNP Table” was filtered by (“Chr” = Y) using filter icon.</li>
<li>Using the 3rd and 7th SNP (index 3010 and 3014, 3 samples had missing values for 3010 and sex was determined to be female using 3014), males were selected as those with high intensities (females have no Y chromosome) and set to have an aux value of 101 by right clicking the selected samples in the “Samples Table”. Similarly, females were selected and set to have an aux value of 102.</li>
<li>“Samples Table” was filtered to have (“Aux” &gt; 100).</li>
<li>“Samples Table” was sorted on “Aux” column and females were selected (value 102) and excluded. Updates were ignored.</li>
<li>In “SNP Table”, all SNPs were selected and Y-snps were clustered (right click and “Cluster Selected SNPs”) on only male samples. Updates were ignored.</li>
<li>Females were re-added to the project in the “Samples Table”.</li>
<li>“SNP Table” was filtered by (“Chr” = X) using filter icon.</li>
<li>In “Samples Table”, males were selected (Aux value 101) and excluded. Updates were ignored.</li>
<li>In “SNP Table”, all SNPs were selected and X-SNPs were clustered (right click and “Cluster Selected SNPs”) on only female samples. Updates were ignored.</li>
<li>Males were re-added to the project in the “Samples Table”.</li>
<li>“SNP Table” was filtered by [ !(“Chr” = X ) AND !(“Chr” = Y ) ], using filter icon.</li>
<li>In “SNP Table”, all SNPs were selected and autosomal SNPs were clustered (right click and “Cluster Selected SNPs”) on all good quality samples. Update SNP statistics.</li>
<li>In the samples table (bottom left panel) call rate was recalculated by clicking “Calculate” (calculator icon) to recalculate SNP call rates with new cluster positions. SNP statistics and heritability estimate updates were ignored.</li>
<li>SNP statistics were updated (“Analysis” &gt; “Update SNP statistics”).</li>
</ol></li>
<li>Review and edit: use filters and scores to evaluate SNPs, correct or zero SNPs as needed. No manual editing was performed.</li>
</ol></li>
<li>New genome studio was exported to the following location: <code>L:\PCN\UBPC\ANALYSES_RECHERCHE\Jenny\PSYMETAB_GWAS</code>, and named: <code>GS_project_26092019_cluster</code>.</li>
<li>Remove filtered individuals:
<ul>
<li>In samples table, remove filter for “Aux” &gt; 100.</li>
<li>Recalculate SNP statistics for these 8 samples only (to save time).</li>
<li>Note that sample call rates have increased with new clustering positions.</li>
</ul></li>
<li>Project was exported as PLINK file to the following location: <code>L:\PCN\UBPC\ANALYSES_RECHERCHE\Jenny\PSYMETAB_GWAS</code>, and named: <code>PSYMETAB_plink_export</code>.</li>
<li>Individual PLINK files within above folder were named according to parent directory as: DATA.ped and DATA.map</li>
<li>PLINK files were then copied to SGG servers using FileZilla: <code>/data/sgg2/jenny/projects/PSYMETAB_GWAS/data/raw</code>.</li>
</ul>
</div>
<div id="notes-and-updates" class="section level2">
<h2>Notes and Updates</h2>
<ul>
<li>Initial data was received on April 8, 2019 and final two plates were received on May 17, 2019.</li>
<li>Processing began with initial files.</li>
<li>July 18, 2019 update:
<ul>
<li>It came to our attention that 15 participants were genotyped that did not consent.</li>
<li>This list was sent to Mylene to be removed.</li>
<li>Plates 27 to 29 were re-provided on 08/08/2019 without these 15 individuals (list provided by Severine in email - see scripts/format_QC_input.r for creation of list of IDs to remove in PLINK).</li>
<li>Until new file was provided, these participants were removed using PLINK to avoid any further analysis of these individuals.</li>
<li>The new genomestudio file was copied to: <code>L:\PCN\UBPC\ANALYSES_RECHERCHE\Jenny\PSYMETAB_GWAS\PSYMETAB_GS2\Plates27to29_0819</code>.</li>
<li>The same process above was followed (data opened in GS, cluster positions imported, and data saved to <code>Plates27to29_0819_cluster</code>, and <code>PLINK_270819_0457</code>).</li>
<li>Old files were deleted to remove all data containing these individuals.</li>
<li>Updated files were then copied to PSYMETAB_GS1.</li>
</ul></li>
<li>August 28, 2019 update:
<ul>
<li>Mylene provided one single genome studio file with all samples (excluding the list from Severine).</li>
<li>These files were copied to UPPC folders and custom clustering was re-performed.</li>
<li>These changes are reflected in the above description.</li>
<li>All old files were subsequently deleted to ensure the data from these participants is completely removed from all databases.</li>
<li>As of September 3, 2019, all clustering was complete and final PLINK files (<code>PLINK_030919_0149</code>) were copied to SGG directory (names of plink files according to parent directory: <code>DATA</code>).</li>
</ul></li>
<li>September 6, 2019 update:
<ul>
<li>It was decided that all IDs part of PSYMETAB should be randomized to ensure they are not identifiable.</li>
<li>We had a meeting to discuss (Celine, Fred, Chin, Nermine, and Claire), and decided to use a CHUV program (GPCR) for the randomization process.</li>
<li>We requested with Mylene to create a new project with the new IDs, but she suggested to create our own GS project.</li>
<li>She provided all relevant data to create our own GS project.</li>
<li>The description above reflects these changes.</li>
</ul></li>
<li>As of October 11, all GS file were created, clustered and exported as PLINK files and subsequently moved to the Zoltan’s.</li>
</ul>
</div>
<div id="creating-full-data-table-for-use-in-penncnv" class="section level2">
<h2>Creating full data table for use in penncnv</h2>
<ul>
<li>Instructions to create data for penncnv input were followed <a href="http://penncnv.openbioinformatics.org/en/latest/user-guide/input/">here</a></li>
<li><code>GS_project_26092019_cluster.bsc</code>, located here: <code>L:\PCN\UBPC\ANALYSES_RECHERCHE\Jenny\PSYMETAB_GWAS\GS_project_26092019_cluster</code>, was used to create the full data table.</li>
<li>Output file was named the default name of <code>Full Data Table.txt</code> and saved in the same location.</li>
<li>The output was then moved to the server to and run on with penncnv.</li>
</ul>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
