<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>GWAS</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/readable.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">PSYMETAB</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="setup.html">Setup</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/jennysjaarda/PSYMETAB">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">GWAS</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2020-06-23
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>PSYMETAB/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20191126code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20191126)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20191126code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20191126)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomjennysjaardaPSYMETABtree55b0a65463c2a86fab310892f4f1bb68a6087013targetblank55b0a65a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/jennysjaarda/PSYMETAB/tree/55b0a65463c2a86fab310892f4f1bb68a6087013" target="_blank">55b0a65</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomjennysjaardaPSYMETABtree55b0a65463c2a86fab310892f4f1bb68a6087013targetblank55b0a65a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    ._docs
    Ignored:    .drake/
    Ignored:    analysis/.Rhistory
    Ignored:    analysis/._GWAS.Rmd
    Ignored:    analysis/._data_processing_in_genomestudio.Rmd
    Ignored:    analysis/._quality_control.Rmd
    Ignored:    analysis/GWAS/
    Ignored:    analysis/PRS/
    Ignored:    analysis/QC/
    Ignored:    analysis/site_libs/
    Ignored:    analysis_prep_1_clustermq.out
    Ignored:    analysis_prep_2_clustermq.out
    Ignored:    analysis_prep_3_clustermq.out
    Ignored:    analysis_prep_4_clustermq.out
    Ignored:    data/processed/
    Ignored:    data/raw/
    Ignored:    download_impute_1_clustermq.out
    Ignored:    init_analysis_1_clustermq.out
    Ignored:    init_analysis_2_clustermq.out
    Ignored:    init_analysis_3_clustermq.out
    Ignored:    init_analysis_4_clustermq.out
    Ignored:    init_analysis_5_clustermq.out
    Ignored:    init_analysis_6_clustermq.out
    Ignored:    packrat/lib-R/
    Ignored:    packrat/lib-ext/
    Ignored:    packrat/lib/
    Ignored:    post_impute_1_clustermq.out
    Ignored:    pre_impute_qc_1_clustermq.out
    Ignored:    process_init_10_clustermq.out
    Ignored:    process_init_11_clustermq.out
    Ignored:    process_init_12_clustermq.out
    Ignored:    process_init_13_clustermq.out
    Ignored:    process_init_14_clustermq.out
    Ignored:    process_init_15_clustermq.out
    Ignored:    process_init_16_clustermq.out
    Ignored:    process_init_17_clustermq.out
    Ignored:    process_init_18_clustermq.out
    Ignored:    process_init_19_clustermq.out
    Ignored:    process_init_1_clustermq.out
    Ignored:    process_init_20_clustermq.out
    Ignored:    process_init_21_clustermq.out
    Ignored:    process_init_22_clustermq.out
    Ignored:    process_init_23_clustermq.out
    Ignored:    process_init_24_clustermq.out
    Ignored:    process_init_25_clustermq.out
    Ignored:    process_init_26_clustermq.out
    Ignored:    process_init_27_clustermq.out
    Ignored:    process_init_28_clustermq.out
    Ignored:    process_init_29_clustermq.out
    Ignored:    process_init_2_clustermq.out
    Ignored:    process_init_30_clustermq.out
    Ignored:    process_init_31_clustermq.out
    Ignored:    process_init_3_clustermq.out
    Ignored:    process_init_4_clustermq.out
    Ignored:    process_init_5_clustermq.out
    Ignored:    process_init_6_clustermq.out
    Ignored:    process_init_7_clustermq.out
    Ignored:    process_init_8_clustermq.out
    Ignored:    process_init_9_clustermq.out
    Ignored:    prs_1_clustermq.out
    Ignored:    prs_2_clustermq.out
    Ignored:    prs_3_clustermq.out
    Ignored:    prs_4_clustermq.out

Untracked files:
    Untracked:  analysis/genetic_quality_control.Rmd
    Untracked:  analysis/plans.Rmd
    Untracked:  analysis_prep.log
    Untracked:  download_impute.log
    Untracked:  grs.log
    Untracked:  init_analysis.log
    Untracked:  process_init.log
    Untracked:  prs.log

Unstaged changes:
    Modified:   analysis/GWAS.Rmd
    Modified:   analysis/data_sources.Rmd
    Modified:   analysis/index.Rmd
    Modified:   analysis/pheno_quality_control.Rmd
    Deleted:    analysis/project.Rmd
    Modified:   analysis/quality_control.Rmd
    Modified:   cache_log.csv
    Modified:   post_impute.log
    Modified:   slurm_clustermq.tmpl

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the R Markdown and HTML files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view them.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jennysjaarda/PSYMETAB/blob/a56ae32bbe88d700370c71dda4c6ffcedfb4a55c/analysis/GWAS.Rmd" target="_blank">a56ae32</a>
</td>
<td>
Sjaarda Jennifer Lynn
</td>
<td>
2020-01-22
</td>
<td>
add GWAS functions
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jennysjaarda/PSYMETAB/46477dd771ba63ffa5d44a0d4f53a1a835a71c87/docs/GWAS.html" target="_blank">46477dd</a>
</td>
<td>
Jenny Sjaarda
</td>
<td>
2019-12-06
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jennysjaarda/PSYMETAB/blob/16f6c28fc976c5b7dc5ac37ba910ec89089f112a/analysis/GWAS.Rmd" target="_blank">16f6c28</a>
</td>
<td>
Sjaarda Jennifer Lynn
</td>
<td>
2019-12-06
</td>
<td>
add gwas Rmd
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p><strong>The following document outlines and summarizes the GWAS analyses run in PSYMETAB.</strong></p>
<div id="analysis-preparation." class="section level1">
<h1>Analysis preparation.</h1>
<p>found that GYYEHMDR is listed as both Male and Female -&gt; should be female only.</p>
</div>
<div id="gwas." class="section level1">
<h1>GWAS.</h1>
<div id="general-settings." class="section level2">
<h2>General settings.</h2>
<div id="drug-classification." class="section level3">
<h3>Drug classification.</h3>
<p>After discussion with others, it was decided to define drugs into three classes</p>
<ol style="list-style-type: decimal">
<li><p>AP1 pour faible risque de prise de poids: Amisulpride, Aripiprazole, Brexpiprazole, Cariprazine, Carbamazepine, Chlorprothixene, Flupentixol, Fluphenazine, Haloperidol, Lurasidone, Pipampérone, Promazine, Sertindole, Sulpiride, Tiapride</p></li>
<li><p>AP2 pour risque intermédiaire de prise de poids: Amitriptyline, Asenapine, Clomipramine, Dibenzepine, Doxepine, Imipramine, Lévomépromazine, Lithium, Mirtazapine, Nortriptyline, Opipramol, Palipéridone, Quétiapine, Rispéridone, Trimipramine, Zuclopenthixol</p></li>
<li><p>AP3 pour risque élevé de prise de poids: Clozapine, Olanzapine, Valproate</p></li>
</ol>
<p>Specifically, high, medium and low inducers were defined as follows:</p>
<pre class="r"><code>high_inducers</code></pre>
<pre><code>[1] &quot;Olanzapine&quot; &quot;Clozapine&quot;  &quot;Valproate&quot; </code></pre>
<pre class="r"><code>med_inducers</code></pre>
<pre><code> [1] &quot;Amitriptyline&quot;   &quot;Asenapine&quot;       &quot;Clomipramine&quot;    &quot;Doxepine&quot;       
 [5] &quot;Levomepromazine&quot; &quot;Lithium&quot;         &quot;Mirtazapine&quot;     &quot;Paliperidone&quot;   
 [9] &quot;Risperidone&quot;     &quot;Quetiapine&quot;      &quot;Trimipramine&quot;    &quot;Zuclopenthixol&quot; </code></pre>
<pre class="r"><code>low_inducers</code></pre>
<pre><code> [1] &quot;Amisulpride&quot;     &quot;Aripiprazole&quot;    &quot;Brexpiprazole&quot;   &quot;Cariprazine&quot;    
 [5] &quot;Carbamazepine&quot;   &quot;Chlorprothixene&quot; &quot;Flupentixol&quot;     &quot;Fluphenazine&quot;   
 [9] &quot;Haloperidol&quot;     &quot;Lurasidone&quot;      &quot;Pipamperone&quot;     &quot;Sertindole&quot;     
[13] &quot;Sulpiride&quot;       &quot;Tiapride&quot;       </code></pre>
</div>
<div id="check-if-mois-column-makes-sense." class="section level3">
<h3>Check if “Mois” column makes sense.</h3>
<p>Plus or minus on follow-up to double check that the “Mois” column make sense.</p>
<p>leeway_time &lt;- 90 For example, if one visit is marked month 3 on September 5/15; and the previous visit marked month 0 on June 15/19, the script below will check if: the difference between the two dates (September 5/15 - June 15/19) is within the difference between the two months (3 = 90 days) +/- the ‘leeway_time’</p>
<p>min_follow_up &lt;- 14</p>
<p>Three types of GWAS were run</p>
</div>
</div>
<div id="interaction" class="section level2">
<h2>Interaction</h2>
</div>
<div id="baseline" class="section level2">
<h2>Baseline</h2>
</div>
<div id="subgroup" class="section level2">
<h2>Subgroup</h2>
<pre class="r"><code>het_out &lt;-
  fread(
    paste0(&quot;analysis/GWAS/subgroup/&quot;, eth, &quot;/&quot;, output, &quot;_&quot;, suffix,
      &quot;_&quot;, eth, &quot;.&quot;, outcome_var, &quot;.het&quot;),
    header = F,
    data.table = F
  )

sig_list &lt;- which(as.numeric(het_out[, 2]) &lt; 5e-08)
length(sig_list)
colnames(het_out) &lt;- c(&quot;ID&quot;, &quot;P_HET&quot;)
result &lt;- cbind(filter, &quot;P_HET&quot; = het_out$P_HET)
png(
  paste0(&quot;analysis/GWAS/subgroup/&quot;, eth, &quot;/&quot;, output, &quot;_&quot;, suffix, &quot;_&quot;, eth, &quot;.&quot;, outcome_var, &quot;.het.manhattan.png&quot;),
  width = 2000,
  height = 1000,
  pointsize = 18
)
manhattan(result, p = &quot;P_HET&quot;, chr = &quot;CHROM&quot;)
dev.off()

png(
  paste0(
    &quot;analysis/GWAS/subgroup/&quot;, eth, &quot;/&quot;, output, &quot;_&quot;, suffix, &quot;_&quot;, eth, &quot;.&quot;, outcome_var, &quot;.het.qq.png&quot;),
  width = 2000,
  height = 1000,
  pointsize = 18
)
qq(result$P_HET)
dev.off()


#### GWAS
    sig_nodrug &lt;-nodrug[ which(nodrug$P &lt; 5e-08),]
    sig_drug &lt;- drug[which(drug$P &lt; 5e-08),]
    for(data in c(&quot;nodrug&quot;, &quot;drug&quot;))
    {
      gwas_result &lt;- get(data)
      joint &lt;- reduce(list(gwas_result,freq,info), full_join, by = &quot;ID&quot;)
      sig &lt;- joint %&gt;%
              mutate_at(&quot;P&quot;, as.numeric) %&gt;%
              filter(P &lt; 5e-06) %&gt;%
              filter(ALT_FREQS &gt; maf_threshold &amp; ALT_FREQS &lt; (1 - maf_threshold)) %&gt;%
              filter(R2 &gt; info_threshold)

      joint_maf &lt;- joint %&gt;% filter(ALT_FREQS &gt; maf_threshold &amp; ALT_FREQS &lt; (1- maf_threshold))%&gt;% mutate_at(&quot;P&quot;, as.numeric)
      sig &lt;- joint_maf  %&gt;% filter(P &lt; gw_sig)
      png(&quot;man_interaction.png&quot;, width=2000, height=1000, pointsize=18)
      manhattan(joint_maf)
      dev.off()

      png(&quot;interaction_qq2.png&quot;, width=2000, height=1000, pointsize=18)
      qq(joint_maf$P)
      dev.off()

            qq(gwas_result2$P)
    # sig_nodrug &lt;- sig
    }
    
    gwas_result &lt;- fread(result, data.table=F, stringsAsFactors=F)
  gwas_result2 &lt;- gwas_result %&gt;% rename(CHR = &quot;#CHROM&quot;) %&gt;% rename(BP = POS) %&gt;% filter(!is.na(P))


  #### GWAS
  sig_nodrug &lt;-nodrug[ which(nodrug$P &lt; 5e-08),]
  sig_drug &lt;- drug[which(drug$P &lt; 5e-08),]

  for(data in c(&quot;nodrug&quot;, &quot;drug&quot;))
  {
  gwas_result &lt;- get(data)
  gwas_munge &lt;- gwas_result %&gt;% rename(CHR = &quot;#CHROM&quot;) %&gt;% rename(BP = POS) %&gt;% filter(!is.na(P))
  joint &lt;- reduce(list(gwas_munge,freq,info), full_join, by = &quot;ID&quot;)
  sig &lt;- joint %&gt;%
          mutate_at(&quot;P&quot;, as.numeric) %&gt;%
          filter(P &lt; 5e-06) %&gt;%
          filter(ALT_FREQS &gt; maf_threshold &amp; ALT_FREQS &lt; (1 - maf_threshold)) %&gt;%
          filter(R2 &gt; info_threshold)
  # sig_nodrug &lt;- sig
  }</code></pre>
</div>
<div id="adjustment-co-variants" class="section level2">
<h2>Adjustment co-variants</h2>
<pre class="r"><code>loadd(pheno_raw)

# temp &lt;-   pheno_raw %&gt;%
#     mutate(Date = as.Date(Date, format = &#39;%d.%m.%Y&#39;))  %&gt;% filter(!is.na(Date)) %&gt;% arrange(Date)  %&gt;%
#     mutate(AP1 = gsub(&quot; &quot;, &quot;_&quot;,AP1)) %&gt;% mutate_at(&quot;AP1&quot;,as.factor) %&gt;% mutate(AP1 = gsub(&quot;_.*$&quot;,&quot;&quot;, AP1)) %&gt;% mutate(AP1 = na_if(AP1, &quot;&quot;)) %&gt;%## merge retard/depot with original
#     group_by(GEN) %&gt;%  mutate(sex = check_sex(Sexe))
# temp %&gt;% filter(is.na(sex)) %&gt;% dplyr::select(sex, Sexe) %&gt;% filter(!is.na(Sexe))
#
# GEN      sex   Sexe
#   &lt;chr&gt;    &lt;chr&gt; &lt;chr&gt;
# 1 GYYEHMDR NA    M
# 2 GYYEHMDR NA    M
# 3 GYYEHMDR NA    M
# 4 GYYEHMDR NA    F
# 5 GYYEHMDR NA    F</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.5.3 (2019-03-11)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

Matrix products: default
BLAS: /data/sgg2/jenny/bin/R-3.5.3/lib64/R/lib/libRblas.so
LAPACK: /data/sgg2/jenny/bin/R-3.5.3/lib64/R/lib/libRlapack.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] tidylog_1.0.1           OpenImageR_1.1.6        fuzzyjoin_0.1.5        
 [4] kableExtra_1.1.0        R.utils_2.9.2           R.oo_1.23.0            
 [7] R.methodsS3_1.7.1       TwoSampleMR_0.4.25      reader_1.0.6           
[10] NCmisc_1.1.6            optparse_1.6.4          readxl_1.3.1           
[13] ggthemes_4.2.0          tryCatchLog_1.1.6       futile.logger_1.4.3    
[16] DataExplorer_0.8.0      taRifx_1.0.6.1          qqman_0.1.4            
[19] MASS_7.3-51.5           bit64_0.9-7             bit_1.1-14             
[22] rslurm_0.5.0            rmeta_3.0               devtools_2.2.1         
[25] usethis_1.5.1           data.table_1.12.8       clustermq_0.8.8.1      
[28] future.batchtools_0.8.1 future_1.15.1           rlang_0.4.5            
[31] knitr_1.26              drake_7.12.0.9000       forcats_0.4.0          
[34] stringr_1.4.0           dplyr_0.8.3             purrr_0.3.3            
[37] readr_1.3.1             tidyr_1.0.3             tibble_2.1.3           
[40] ggplot2_3.2.1           tidyverse_1.3.0         pacman_0.5.1           
[43] processx_3.4.1          workflowr_1.6.0        

loaded via a namespace (and not attached):
  [1] backports_1.1.6      plyr_1.8.5           igraph_1.2.5        
  [4] lazyeval_0.2.2       storr_1.2.1          listenv_0.8.0       
  [7] digest_0.6.25        htmltools_0.4.0      tiff_0.1-5          
 [10] fansi_0.4.1          magrittr_1.5         checkmate_1.9.4     
 [13] memoise_1.1.0        base64url_1.4        remotes_2.1.0       
 [16] globals_0.12.5       modelr_0.1.5         prettyunits_1.1.0   
 [19] jpeg_0.1-8.1         colorspace_1.4-1     rvest_0.3.5         
 [22] rappdirs_0.3.1       haven_2.2.0          xfun_0.11           
 [25] callr_3.4.0          crayon_1.3.4         jsonlite_1.6        
 [28] brew_1.0-6           glue_1.4.0           gtable_0.3.0        
 [31] webshot_0.5.2        pkgbuild_1.0.6       scales_1.1.0        
 [34] futile.options_1.0.1 DBI_1.1.0            Rcpp_1.0.3          
 [37] xtable_1.8-4         viridisLite_0.3.0    progress_1.2.2      
 [40] txtq_0.2.0           clisymbols_1.2.0     htmlwidgets_1.5.1   
 [43] httr_1.4.1           getopt_1.20.3        calibrate_1.7.5     
 [46] ellipsis_0.3.0       pkgconfig_2.0.3      dbplyr_1.4.2        
 [49] tidyselect_0.2.5     reshape2_1.4.3       later_1.0.0         
 [52] munsell_0.5.0        cellranger_1.1.0     tools_3.5.3         
 [55] cli_2.0.1            generics_0.0.2       broom_0.5.3         
 [58] fastmap_1.0.1        evaluate_0.14        yaml_2.2.0          
 [61] fs_1.3.1             packrat_0.5.0        nlme_3.1-143        
 [64] mime_0.8             whisker_0.4          formatR_1.7         
 [67] proftools_0.99-2     xml2_1.2.2           compiler_3.5.3      
 [70] rstudioapi_0.10      png_0.1-7            filelock_1.0.2      
 [73] testthat_2.3.1       reprex_0.3.0         stringi_1.4.5       
 [76] ps_1.3.0             desc_1.2.0           lattice_0.20-38     
 [79] vctrs_0.2.4          pillar_1.4.3         lifecycle_0.1.0     
 [82] networkD3_0.4        httpuv_1.5.2         R6_2.4.1            
 [85] promises_1.1.0       gridExtra_2.3        sessioninfo_1.1.1   
 [88] codetools_0.2-16     lambda.r_1.2.4       assertthat_0.2.1    
 [91] pkgload_1.0.2        rprojroot_1.3-2      withr_2.1.2         
 [94] batchtools_0.9.12    parallel_3.5.3       hms_0.5.3           
 [97] grid_3.5.3           rmarkdown_1.18       git2r_0.26.1        
[100] shiny_1.4.0          lubridate_1.7.4     </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
