<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Home</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/readable.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">PSYMETAB</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="setup.html">Setup</a>
</li>
<li>
  <a href="plans.html">Analysis plans</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/jennysjaarda/PSYMETAB">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Home</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report"
  data-toggle="collapse" data-target="#workflowr-report">
  
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>

<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
  <li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
  <li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
  <li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>

<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
  <p>
<strong>Last updated:</strong> 2021-05-31
</p>
  <p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
  <p>
<strong>Knit directory:</strong> <code>PSYMETAB/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
  <p>
  
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
  <div class="panel-group" id="workflowr-checks">
  <div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
  
<p>The R Markdown file has unstaged changes. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
  
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20191126code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20191126)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20191126code" class="panel-collapse collapse">
<div class="panel-body">
  
<p>The command <code>set.seed(20191126)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
  
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
  
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
  
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomjennysjaardaPSYMETABtreef5126df40b07977a0af2a0b0197cd8909c17af58targetblankf5126dfa"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/jennysjaarda/PSYMETAB/tree/f5126df40b07977a0af2a0b0197cd8909c17af58" target="_blank">f5126df</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomjennysjaardaPSYMETABtreef5126df40b07977a0af2a0b0197cd8909c17af58targetblankf5126dfa" class="panel-collapse collapse">
<div class="panel-body">
  

<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
 <pre><code>
Ignored files:
    Ignored:    ._docs
    Ignored:    .drake/
    Ignored:    analysis/.Rhistory
    Ignored:    analysis/._GWAS.Rmd
    Ignored:    analysis/._data_processing_in_genomestudio.Rmd
    Ignored:    analysis/._quality_control.Rmd
    Ignored:    analysis/GWAS/
    Ignored:    analysis/PRS/
    Ignored:    analysis/QC/
    Ignored:    analysis/Rlogo2.png
    Ignored:    analysis/_drake.R
    Ignored:    analysis/figure/
    Ignored:    analysis/rplot.jpg
    Ignored:    analysis_prep.log
    Ignored:    analysis_prep_10_clustermq.out
    Ignored:    analysis_prep_11_clustermq.out
    Ignored:    analysis_prep_12_clustermq.out
    Ignored:    analysis_prep_13_clustermq.out
    Ignored:    analysis_prep_14_clustermq.out
    Ignored:    analysis_prep_15_clustermq.out
    Ignored:    analysis_prep_16_clustermq.out
    Ignored:    analysis_prep_17_clustermq.out
    Ignored:    analysis_prep_18_clustermq.out
    Ignored:    analysis_prep_19_clustermq.out
    Ignored:    analysis_prep_1_clustermq.out
    Ignored:    analysis_prep_20_clustermq.out
    Ignored:    analysis_prep_21_clustermq.out
    Ignored:    analysis_prep_22_clustermq.out
    Ignored:    analysis_prep_23_clustermq.out
    Ignored:    analysis_prep_24_clustermq.out
    Ignored:    analysis_prep_25_clustermq.out
    Ignored:    analysis_prep_26_clustermq.out
    Ignored:    analysis_prep_27_clustermq.out
    Ignored:    analysis_prep_28_clustermq.out
    Ignored:    analysis_prep_29_clustermq.out
    Ignored:    analysis_prep_2_clustermq.out
    Ignored:    analysis_prep_30_clustermq.out
    Ignored:    analysis_prep_31_clustermq.out
    Ignored:    analysis_prep_32_clustermq.out
    Ignored:    analysis_prep_33_clustermq.out
    Ignored:    analysis_prep_34_clustermq.out
    Ignored:    analysis_prep_35_clustermq.out
    Ignored:    analysis_prep_36_clustermq.out
    Ignored:    analysis_prep_37_clustermq.out
    Ignored:    analysis_prep_38_clustermq.out
    Ignored:    analysis_prep_39_clustermq.out
    Ignored:    analysis_prep_3_clustermq.out
    Ignored:    analysis_prep_40_clustermq.out
    Ignored:    analysis_prep_4_clustermq.out
    Ignored:    analysis_prep_5_clustermq.out
    Ignored:    analysis_prep_6_clustermq.out
    Ignored:    analysis_prep_7_clustermq.out
    Ignored:    analysis_prep_8_clustermq.out
    Ignored:    analysis_prep_9_clustermq.out
    Ignored:    data/processed/
    Ignored:    data/raw/
    Ignored:    download_impute.log
    Ignored:    extract_sig.log
    Ignored:    grs.log
    Ignored:    init_analysis.log
    Ignored:    init_analysis_10_clustermq.out
    Ignored:    init_analysis_11_clustermq.out
    Ignored:    init_analysis_12_clustermq.out
    Ignored:    init_analysis_13_clustermq.out
    Ignored:    init_analysis_14_clustermq.out
    Ignored:    init_analysis_15_clustermq.out
    Ignored:    init_analysis_16_clustermq.out
    Ignored:    init_analysis_17_clustermq.out
    Ignored:    init_analysis_18_clustermq.out
    Ignored:    init_analysis_19_clustermq.out
    Ignored:    init_analysis_1_clustermq.out
    Ignored:    init_analysis_20_clustermq.out
    Ignored:    init_analysis_2_clustermq.out
    Ignored:    init_analysis_3_clustermq.out
    Ignored:    init_analysis_4_clustermq.out
    Ignored:    init_analysis_5_clustermq.out
    Ignored:    init_analysis_6_clustermq.out
    Ignored:    init_analysis_7_clustermq.out
    Ignored:    init_analysis_8_clustermq.out
    Ignored:    init_analysis_9_clustermq.out
    Ignored:    packrat/lib-R/
    Ignored:    packrat/lib-ext/
    Ignored:    packrat/lib/
    Ignored:    plan.log
    Ignored:    plan_100_clustermq.out
    Ignored:    plan_101_clustermq.out
    Ignored:    plan_102_clustermq.out
    Ignored:    plan_103_clustermq.out
    Ignored:    plan_104_clustermq.out
    Ignored:    plan_105_clustermq.out
    Ignored:    plan_106_clustermq.out
    Ignored:    plan_107_clustermq.out
    Ignored:    plan_108_clustermq.out
    Ignored:    plan_109_clustermq.out
    Ignored:    plan_10_clustermq.out
    Ignored:    plan_110_clustermq.out
    Ignored:    plan_111_clustermq.out
    Ignored:    plan_112_clustermq.out
    Ignored:    plan_113_clustermq.out
    Ignored:    plan_114_clustermq.out
    Ignored:    plan_115_clustermq.out
    Ignored:    plan_116_clustermq.out
    Ignored:    plan_117_clustermq.out
    Ignored:    plan_118_clustermq.out
    Ignored:    plan_119_clustermq.out
    Ignored:    plan_11_clustermq.out
    Ignored:    plan_120_clustermq.out
    Ignored:    plan_12_clustermq.out
    Ignored:    plan_13_clustermq.out
    Ignored:    plan_14_clustermq.out
    Ignored:    plan_15_clustermq.out
    Ignored:    plan_16_clustermq.out
    Ignored:    plan_17_clustermq.out
    Ignored:    plan_18_clustermq.out
    Ignored:    plan_19_clustermq.out
    Ignored:    plan_1_clustermq.out
    Ignored:    plan_20_clustermq.out
    Ignored:    plan_21_clustermq.out
    Ignored:    plan_22_clustermq.out
    Ignored:    plan_23_clustermq.out
    Ignored:    plan_24_clustermq.out
    Ignored:    plan_25_clustermq.out
    Ignored:    plan_26_clustermq.out
    Ignored:    plan_27_clustermq.out
    Ignored:    plan_28_clustermq.out
    Ignored:    plan_29_clustermq.out
    Ignored:    plan_2_clustermq.out
    Ignored:    plan_30_clustermq.out
    Ignored:    plan_31_clustermq.out
    Ignored:    plan_32_clustermq.out
    Ignored:    plan_33_clustermq.out
    Ignored:    plan_34_clustermq.out
    Ignored:    plan_35_clustermq.out
    Ignored:    plan_36_clustermq.out
    Ignored:    plan_37_clustermq.out
    Ignored:    plan_38_clustermq.out
    Ignored:    plan_39_clustermq.out
    Ignored:    plan_3_clustermq.out
    Ignored:    plan_40_clustermq.out
    Ignored:    plan_41_clustermq.out
    Ignored:    plan_42_clustermq.out
    Ignored:    plan_43_clustermq.out
    Ignored:    plan_44_clustermq.out
    Ignored:    plan_45_clustermq.out
    Ignored:    plan_46_clustermq.out
    Ignored:    plan_47_clustermq.out
    Ignored:    plan_48_clustermq.out
    Ignored:    plan_49_clustermq.out
    Ignored:    plan_4_clustermq.out
    Ignored:    plan_50_clustermq.out
    Ignored:    plan_51_clustermq.out
    Ignored:    plan_52_clustermq.out
    Ignored:    plan_53_clustermq.out
    Ignored:    plan_54_clustermq.out
    Ignored:    plan_55_clustermq.out
    Ignored:    plan_56_clustermq.out
    Ignored:    plan_57_clustermq.out
    Ignored:    plan_58_clustermq.out
    Ignored:    plan_59_clustermq.out
    Ignored:    plan_5_clustermq.out
    Ignored:    plan_60_clustermq.out
    Ignored:    plan_61_clustermq.out
    Ignored:    plan_62_clustermq.out
    Ignored:    plan_63_clustermq.out
    Ignored:    plan_64_clustermq.out
    Ignored:    plan_65_clustermq.out
    Ignored:    plan_66_clustermq.out
    Ignored:    plan_67_clustermq.out
    Ignored:    plan_68_clustermq.out
    Ignored:    plan_69_clustermq.out
    Ignored:    plan_6_clustermq.out
    Ignored:    plan_70_clustermq.out
    Ignored:    plan_71_clustermq.out
    Ignored:    plan_72_clustermq.out
    Ignored:    plan_73_clustermq.out
    Ignored:    plan_74_clustermq.out
    Ignored:    plan_75_clustermq.out
    Ignored:    plan_76_clustermq.out
    Ignored:    plan_77_clustermq.out
    Ignored:    plan_78_clustermq.out
    Ignored:    plan_79_clustermq.out
    Ignored:    plan_7_clustermq.out
    Ignored:    plan_80_clustermq.out
    Ignored:    plan_81_clustermq.out
    Ignored:    plan_82_clustermq.out
    Ignored:    plan_83_clustermq.out
    Ignored:    plan_84_clustermq.out
    Ignored:    plan_85_clustermq.out
    Ignored:    plan_86_clustermq.out
    Ignored:    plan_87_clustermq.out
    Ignored:    plan_88_clustermq.out
    Ignored:    plan_89_clustermq.out
    Ignored:    plan_8_clustermq.out
    Ignored:    plan_90_clustermq.out
    Ignored:    plan_91_clustermq.out
    Ignored:    plan_92_clustermq.out
    Ignored:    plan_93_clustermq.out
    Ignored:    plan_94_clustermq.out
    Ignored:    plan_95_clustermq.out
    Ignored:    plan_96_clustermq.out
    Ignored:    plan_97_clustermq.out
    Ignored:    plan_98_clustermq.out
    Ignored:    plan_99_clustermq.out
    Ignored:    plan_9_clustermq.out
    Ignored:    process_init.log
    Ignored:    process_init_100_clustermq.out
    Ignored:    process_init_101_clustermq.out
    Ignored:    process_init_102_clustermq.out
    Ignored:    process_init_103_clustermq.out
    Ignored:    process_init_104_clustermq.out
    Ignored:    process_init_105_clustermq.out
    Ignored:    process_init_106_clustermq.out
    Ignored:    process_init_107_clustermq.out
    Ignored:    process_init_108_clustermq.out
    Ignored:    process_init_109_clustermq.out
    Ignored:    process_init_10_clustermq.out
    Ignored:    process_init_110_clustermq.out
    Ignored:    process_init_111_clustermq.out
    Ignored:    process_init_112_clustermq.out
    Ignored:    process_init_113_clustermq.out
    Ignored:    process_init_114_clustermq.out
    Ignored:    process_init_115_clustermq.out
    Ignored:    process_init_116_clustermq.out
    Ignored:    process_init_117_clustermq.out
    Ignored:    process_init_118_clustermq.out
    Ignored:    process_init_119_clustermq.out
    Ignored:    process_init_11_clustermq.out
    Ignored:    process_init_120_clustermq.out
    Ignored:    process_init_12_clustermq.out
    Ignored:    process_init_13_clustermq.out
    Ignored:    process_init_14_clustermq.out
    Ignored:    process_init_15_clustermq.out
    Ignored:    process_init_16_clustermq.out
    Ignored:    process_init_17_clustermq.out
    Ignored:    process_init_18_clustermq.out
    Ignored:    process_init_19_clustermq.out
    Ignored:    process_init_1_clustermq.out
    Ignored:    process_init_20_clustermq.out
    Ignored:    process_init_21_clustermq.out
    Ignored:    process_init_22_clustermq.out
    Ignored:    process_init_23_clustermq.out
    Ignored:    process_init_24_clustermq.out
    Ignored:    process_init_25_clustermq.out
    Ignored:    process_init_26_clustermq.out
    Ignored:    process_init_27_clustermq.out
    Ignored:    process_init_28_clustermq.out
    Ignored:    process_init_29_clustermq.out
    Ignored:    process_init_2_clustermq.out
    Ignored:    process_init_30_clustermq.out
    Ignored:    process_init_31_clustermq.out
    Ignored:    process_init_32_clustermq.out
    Ignored:    process_init_33_clustermq.out
    Ignored:    process_init_34_clustermq.out
    Ignored:    process_init_35_clustermq.out
    Ignored:    process_init_36_clustermq.out
    Ignored:    process_init_37_clustermq.out
    Ignored:    process_init_38_clustermq.out
    Ignored:    process_init_39_clustermq.out
    Ignored:    process_init_3_clustermq.out
    Ignored:    process_init_40_clustermq.out
    Ignored:    process_init_41_clustermq.out
    Ignored:    process_init_42_clustermq.out
    Ignored:    process_init_43_clustermq.out
    Ignored:    process_init_44_clustermq.out
    Ignored:    process_init_45_clustermq.out
    Ignored:    process_init_46_clustermq.out
    Ignored:    process_init_47_clustermq.out
    Ignored:    process_init_48_clustermq.out
    Ignored:    process_init_49_clustermq.out
    Ignored:    process_init_4_clustermq.out
    Ignored:    process_init_50_clustermq.out
    Ignored:    process_init_51_clustermq.out
    Ignored:    process_init_52_clustermq.out
    Ignored:    process_init_53_clustermq.out
    Ignored:    process_init_54_clustermq.out
    Ignored:    process_init_55_clustermq.out
    Ignored:    process_init_56_clustermq.out
    Ignored:    process_init_57_clustermq.out
    Ignored:    process_init_58_clustermq.out
    Ignored:    process_init_59_clustermq.out
    Ignored:    process_init_5_clustermq.out
    Ignored:    process_init_60_clustermq.out
    Ignored:    process_init_61_clustermq.out
    Ignored:    process_init_62_clustermq.out
    Ignored:    process_init_63_clustermq.out
    Ignored:    process_init_64_clustermq.out
    Ignored:    process_init_65_clustermq.out
    Ignored:    process_init_66_clustermq.out
    Ignored:    process_init_67_clustermq.out
    Ignored:    process_init_68_clustermq.out
    Ignored:    process_init_69_clustermq.out
    Ignored:    process_init_6_clustermq.out
    Ignored:    process_init_70_clustermq.out
    Ignored:    process_init_71_clustermq.out
    Ignored:    process_init_72_clustermq.out
    Ignored:    process_init_73_clustermq.out
    Ignored:    process_init_74_clustermq.out
    Ignored:    process_init_75_clustermq.out
    Ignored:    process_init_76_clustermq.out
    Ignored:    process_init_77_clustermq.out
    Ignored:    process_init_78_clustermq.out
    Ignored:    process_init_79_clustermq.out
    Ignored:    process_init_7_clustermq.out
    Ignored:    process_init_80_clustermq.out
    Ignored:    process_init_81_clustermq.out
    Ignored:    process_init_82_clustermq.out
    Ignored:    process_init_83_clustermq.out
    Ignored:    process_init_84_clustermq.out
    Ignored:    process_init_85_clustermq.out
    Ignored:    process_init_86_clustermq.out
    Ignored:    process_init_87_clustermq.out
    Ignored:    process_init_88_clustermq.out
    Ignored:    process_init_89_clustermq.out
    Ignored:    process_init_8_clustermq.out
    Ignored:    process_init_90_clustermq.out
    Ignored:    process_init_91_clustermq.out
    Ignored:    process_init_92_clustermq.out
    Ignored:    process_init_93_clustermq.out
    Ignored:    process_init_94_clustermq.out
    Ignored:    process_init_95_clustermq.out
    Ignored:    process_init_96_clustermq.out
    Ignored:    process_init_97_clustermq.out
    Ignored:    process_init_98_clustermq.out
    Ignored:    process_init_99_clustermq.out
    Ignored:    process_init_9_clustermq.out
    Ignored:    process_ukbb.log
    Ignored:    process_ukbb_10_clustermq.out
    Ignored:    process_ukbb_11_clustermq.out
    Ignored:    process_ukbb_12_clustermq.out
    Ignored:    process_ukbb_13_clustermq.out
    Ignored:    process_ukbb_14_clustermq.out
    Ignored:    process_ukbb_15_clustermq.out
    Ignored:    process_ukbb_16_clustermq.out
    Ignored:    process_ukbb_17_clustermq.out
    Ignored:    process_ukbb_18_clustermq.out
    Ignored:    process_ukbb_19_clustermq.out
    Ignored:    process_ukbb_1_clustermq.out
    Ignored:    process_ukbb_20_clustermq.out
    Ignored:    process_ukbb_21_clustermq.out
    Ignored:    process_ukbb_22_clustermq.out
    Ignored:    process_ukbb_2_clustermq.out
    Ignored:    process_ukbb_3_clustermq.out
    Ignored:    process_ukbb_4_clustermq.out
    Ignored:    process_ukbb_5_clustermq.out
    Ignored:    process_ukbb_6_clustermq.out
    Ignored:    process_ukbb_7_clustermq.out
    Ignored:    process_ukbb_8_clustermq.out
    Ignored:    process_ukbb_9_clustermq.out
    Ignored:    prs.log
    Ignored:    prs_1_clustermq.out
    Ignored:    prs_2_clustermq.out
    Ignored:    prs_3_clustermq.out
    Ignored:    prs_4_clustermq.out
    Ignored:    prs_5_clustermq.out
    Ignored:    prs_6_clustermq.out
    Ignored:    prs_7_clustermq.out
    Ignored:    prs_8_clustermq.out
    Ignored:    ukbb.log
    Ignored:    ukbb_10_clustermq.out
    Ignored:    ukbb_11_clustermq.out
    Ignored:    ukbb_12_clustermq.out
    Ignored:    ukbb_13_clustermq.out
    Ignored:    ukbb_14_clustermq.out
    Ignored:    ukbb_15_clustermq.out
    Ignored:    ukbb_16_clustermq.out
    Ignored:    ukbb_17_clustermq.out
    Ignored:    ukbb_18_clustermq.out
    Ignored:    ukbb_19_clustermq.out
    Ignored:    ukbb_1_clustermq.out
    Ignored:    ukbb_20_clustermq.out
    Ignored:    ukbb_21_clustermq.out
    Ignored:    ukbb_22_clustermq.out
    Ignored:    ukbb_23_clustermq.out
    Ignored:    ukbb_24_clustermq.out
    Ignored:    ukbb_2_clustermq.out
    Ignored:    ukbb_3_clustermq.out
    Ignored:    ukbb_4_clustermq.out
    Ignored:    ukbb_5_clustermq.out
    Ignored:    ukbb_6_clustermq.out
    Ignored:    ukbb_7_clustermq.out
    Ignored:    ukbb_8_clustermq.out
    Ignored:    ukbb_9_clustermq.out
    Ignored:    ukbb_analysis.log
    Ignored:    ukbb_analysis_100_clustermq.out
    Ignored:    ukbb_analysis_101_clustermq.out
    Ignored:    ukbb_analysis_102_clustermq.out
    Ignored:    ukbb_analysis_103_clustermq.out
    Ignored:    ukbb_analysis_104_clustermq.out
    Ignored:    ukbb_analysis_105_clustermq.out
    Ignored:    ukbb_analysis_106_clustermq.out
    Ignored:    ukbb_analysis_107_clustermq.out
    Ignored:    ukbb_analysis_108_clustermq.out
    Ignored:    ukbb_analysis_109_clustermq.out
    Ignored:    ukbb_analysis_10_clustermq.out
    Ignored:    ukbb_analysis_110_clustermq.out
    Ignored:    ukbb_analysis_111_clustermq.out
    Ignored:    ukbb_analysis_112_clustermq.out
    Ignored:    ukbb_analysis_113_clustermq.out
    Ignored:    ukbb_analysis_114_clustermq.out
    Ignored:    ukbb_analysis_115_clustermq.out
    Ignored:    ukbb_analysis_116_clustermq.out
    Ignored:    ukbb_analysis_117_clustermq.out
    Ignored:    ukbb_analysis_118_clustermq.out
    Ignored:    ukbb_analysis_119_clustermq.out
    Ignored:    ukbb_analysis_11_clustermq.out
    Ignored:    ukbb_analysis_120_clustermq.out
    Ignored:    ukbb_analysis_121_clustermq.out
    Ignored:    ukbb_analysis_122_clustermq.out
    Ignored:    ukbb_analysis_123_clustermq.out
    Ignored:    ukbb_analysis_124_clustermq.out
    Ignored:    ukbb_analysis_125_clustermq.out
    Ignored:    ukbb_analysis_126_clustermq.out
    Ignored:    ukbb_analysis_127_clustermq.out
    Ignored:    ukbb_analysis_128_clustermq.out
    Ignored:    ukbb_analysis_129_clustermq.out
    Ignored:    ukbb_analysis_12_clustermq.out
    Ignored:    ukbb_analysis_130_clustermq.out
    Ignored:    ukbb_analysis_131_clustermq.out
    Ignored:    ukbb_analysis_132_clustermq.out
    Ignored:    ukbb_analysis_133_clustermq.out
    Ignored:    ukbb_analysis_134_clustermq.out
    Ignored:    ukbb_analysis_135_clustermq.out
    Ignored:    ukbb_analysis_136_clustermq.out
    Ignored:    ukbb_analysis_137_clustermq.out
    Ignored:    ukbb_analysis_138_clustermq.out
    Ignored:    ukbb_analysis_139_clustermq.out
    Ignored:    ukbb_analysis_13_clustermq.out
    Ignored:    ukbb_analysis_140_clustermq.out
    Ignored:    ukbb_analysis_141_clustermq.out
    Ignored:    ukbb_analysis_142_clustermq.out
    Ignored:    ukbb_analysis_143_clustermq.out
    Ignored:    ukbb_analysis_144_clustermq.out
    Ignored:    ukbb_analysis_145_clustermq.out
    Ignored:    ukbb_analysis_146_clustermq.out
    Ignored:    ukbb_analysis_147_clustermq.out
    Ignored:    ukbb_analysis_148_clustermq.out
    Ignored:    ukbb_analysis_149_clustermq.out
    Ignored:    ukbb_analysis_14_clustermq.out
    Ignored:    ukbb_analysis_150_clustermq.out
    Ignored:    ukbb_analysis_151_clustermq.out
    Ignored:    ukbb_analysis_152_clustermq.out
    Ignored:    ukbb_analysis_153_clustermq.out
    Ignored:    ukbb_analysis_154_clustermq.out
    Ignored:    ukbb_analysis_155_clustermq.out
    Ignored:    ukbb_analysis_156_clustermq.out
    Ignored:    ukbb_analysis_157_clustermq.out
    Ignored:    ukbb_analysis_158_clustermq.out
    Ignored:    ukbb_analysis_159_clustermq.out
    Ignored:    ukbb_analysis_15_clustermq.out
    Ignored:    ukbb_analysis_160_clustermq.out
    Ignored:    ukbb_analysis_16_clustermq.out
    Ignored:    ukbb_analysis_17_clustermq.out
    Ignored:    ukbb_analysis_18_clustermq.out
    Ignored:    ukbb_analysis_19_clustermq.out
    Ignored:    ukbb_analysis_1_clustermq.out
    Ignored:    ukbb_analysis_20_clustermq.out
    Ignored:    ukbb_analysis_21_clustermq.out
    Ignored:    ukbb_analysis_22_clustermq.out
    Ignored:    ukbb_analysis_23_clustermq.out
    Ignored:    ukbb_analysis_24_clustermq.out
    Ignored:    ukbb_analysis_25_clustermq.out
    Ignored:    ukbb_analysis_26_clustermq.out
    Ignored:    ukbb_analysis_27_clustermq.out
    Ignored:    ukbb_analysis_28_clustermq.out
    Ignored:    ukbb_analysis_29_clustermq.out
    Ignored:    ukbb_analysis_2_clustermq.out
    Ignored:    ukbb_analysis_30_clustermq.out
    Ignored:    ukbb_analysis_31_clustermq.out
    Ignored:    ukbb_analysis_32_clustermq.out
    Ignored:    ukbb_analysis_33_clustermq.out
    Ignored:    ukbb_analysis_34_clustermq.out
    Ignored:    ukbb_analysis_35_clustermq.out
    Ignored:    ukbb_analysis_36_clustermq.out
    Ignored:    ukbb_analysis_37_clustermq.out
    Ignored:    ukbb_analysis_38_clustermq.out
    Ignored:    ukbb_analysis_39_clustermq.out
    Ignored:    ukbb_analysis_3_clustermq.out
    Ignored:    ukbb_analysis_40_clustermq.out
    Ignored:    ukbb_analysis_41_clustermq.out
    Ignored:    ukbb_analysis_42_clustermq.out
    Ignored:    ukbb_analysis_43_clustermq.out
    Ignored:    ukbb_analysis_44_clustermq.out
    Ignored:    ukbb_analysis_45_clustermq.out
    Ignored:    ukbb_analysis_46_clustermq.out
    Ignored:    ukbb_analysis_47_clustermq.out
    Ignored:    ukbb_analysis_48_clustermq.out
    Ignored:    ukbb_analysis_49_clustermq.out
    Ignored:    ukbb_analysis_4_clustermq.out
    Ignored:    ukbb_analysis_50_clustermq.out
    Ignored:    ukbb_analysis_51_clustermq.out
    Ignored:    ukbb_analysis_52_clustermq.out
    Ignored:    ukbb_analysis_53_clustermq.out
    Ignored:    ukbb_analysis_54_clustermq.out
    Ignored:    ukbb_analysis_55_clustermq.out
    Ignored:    ukbb_analysis_56_clustermq.out
    Ignored:    ukbb_analysis_57_clustermq.out
    Ignored:    ukbb_analysis_58_clustermq.out
    Ignored:    ukbb_analysis_59_clustermq.out
    Ignored:    ukbb_analysis_5_clustermq.out
    Ignored:    ukbb_analysis_60_clustermq.out
    Ignored:    ukbb_analysis_61_clustermq.out
    Ignored:    ukbb_analysis_62_clustermq.out
    Ignored:    ukbb_analysis_63_clustermq.out
    Ignored:    ukbb_analysis_64_clustermq.out
    Ignored:    ukbb_analysis_65_clustermq.out
    Ignored:    ukbb_analysis_66_clustermq.out
    Ignored:    ukbb_analysis_67_clustermq.out
    Ignored:    ukbb_analysis_68_clustermq.out
    Ignored:    ukbb_analysis_69_clustermq.out
    Ignored:    ukbb_analysis_6_clustermq.out
    Ignored:    ukbb_analysis_70_clustermq.out
    Ignored:    ukbb_analysis_71_clustermq.out
    Ignored:    ukbb_analysis_72_clustermq.out
    Ignored:    ukbb_analysis_73_clustermq.out
    Ignored:    ukbb_analysis_74_clustermq.out
    Ignored:    ukbb_analysis_75_clustermq.out
    Ignored:    ukbb_analysis_76_clustermq.out
    Ignored:    ukbb_analysis_77_clustermq.out
    Ignored:    ukbb_analysis_78_clustermq.out
    Ignored:    ukbb_analysis_79_clustermq.out
    Ignored:    ukbb_analysis_7_clustermq.out
    Ignored:    ukbb_analysis_80_clustermq.out
    Ignored:    ukbb_analysis_81_clustermq.out
    Ignored:    ukbb_analysis_82_clustermq.out
    Ignored:    ukbb_analysis_83_clustermq.out
    Ignored:    ukbb_analysis_84_clustermq.out
    Ignored:    ukbb_analysis_85_clustermq.out
    Ignored:    ukbb_analysis_86_clustermq.out
    Ignored:    ukbb_analysis_87_clustermq.out
    Ignored:    ukbb_analysis_88_clustermq.out
    Ignored:    ukbb_analysis_89_clustermq.out
    Ignored:    ukbb_analysis_8_clustermq.out
    Ignored:    ukbb_analysis_90_clustermq.out
    Ignored:    ukbb_analysis_91_clustermq.out
    Ignored:    ukbb_analysis_92_clustermq.out
    Ignored:    ukbb_analysis_93_clustermq.out
    Ignored:    ukbb_analysis_94_clustermq.out
    Ignored:    ukbb_analysis_95_clustermq.out
    Ignored:    ukbb_analysis_96_clustermq.out
    Ignored:    ukbb_analysis_97_clustermq.out
    Ignored:    ukbb_analysis_98_clustermq.out
    Ignored:    ukbb_analysis_99_clustermq.out
    Ignored:    ukbb_analysis_9_clustermq.out

Untracked files:
    Untracked:  Rlogo.png
    Untracked:  Rlogo2.png
    Untracked:  Rplots.pdf
    Untracked:  analysis/GWAS_subgroup_results.Rmd
    Untracked:  analysis/meetings.Rmd
    Untracked:  analysis/report.Rmd
    Untracked:  code/extractions/Aurelie09032021_extract.sh
    Untracked:  code/extractions/MarcodePieri_09032021_extract.sh
    Untracked:  code/extractions/MarcodePieri_09032021_proxies_extract.sh
    Untracked:  forest_plot.rds
    Untracked:  output/PSYMETAB_GWAS_UKBB_controls.csv
    Untracked:  output/PSYMETAB_GWAS_baseline_CEU_result.csv
    Untracked:  output/PSYMETAB_GWAS_case_only_CEU_meta_result.csv
    Untracked:  output/PSYMETAB_GWAS_celine_extract.csv
    Untracked:  output/PSYMETAB_GWAS_subgroup_CEU_result.csv
    Untracked:  output/PSYMETAB_inclusion_dates.txt
    Untracked:  output/coffee_consumed_Neale_UKBB_analysis.csv
    Untracked:  rplot.jpg
    Untracked:  temp

Unstaged changes:
    Modified:   R/functions.R
    Modified:   R/plan.R
    Modified:   _drake.R
    Modified:   analysis/GWAS_UKBB_control.Rmd
    Modified:   analysis/GWAS_UKBB_replication.Rmd
    Modified:   analysis/GWAS_case_only.Rmd
    Modified:   analysis/GWAS_method.Rmd
    Modified:   analysis/GWAS_results.Rmd
    Modified:   analysis/_site.yml
    Modified:   analysis/analysis_summary.Rmd
    Modified:   analysis/create_custom_cluster_files.Rmd
    Modified:   analysis/genetic_quality_control.Rmd
    Modified:   analysis/index.Rmd
    Modified:   analysis/pheno_quality_control.Rmd
    Modified:   analysis/plans.Rmd
    Modified:   analysis/setup.Rmd
    Modified:   cache_log.csv
    Modified:   code/extractions/AudeCoumau_extract.sh
    Modified:   code/extractions/AurelieReymond_extract.sh
    Modified:   code/extractions/ChinEap_extract.sh
    Modified:   code/extractions/FlavioFerreira_extract.sh
    Modified:   code/extractions/FredVandenberghe_extract.sh
    Modified:   code/extractions/MarcodePieri_extract.sh
    Modified:   code/extractions/MurielleBrocard_extract.sh
    Modified:   code/extractions/NermineLaaboub_extract.sh
    Modified:   code/extractions/NicolasAnsermot_extract.sh
    Modified:   code/extractions/SeverineCrettolWavre_extract.sh
    Modified:   post_impute.log
    Modified:   slurm_clustermq.tmpl

</code></pre> <p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>

</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
  
<p>
These are the previous versions of the R Markdown and HTML files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view them.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jennysjaarda/PSYMETAB/3cea988925216111aea6b7a3cf16ddc9b156cd22/docs/index.html" target="_blank">3cea988</a>
</td>
<td>
Jenny Sjaarda
</td>
<td>
2021-03-04
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jennysjaarda/PSYMETAB/blob/c96d28fe674dee4f066eaa79587384e0b4f67921/analysis/index.Rmd" target="_blank">c96d28f</a>
</td>
<td>
Jenny Sjaarda
</td>
<td>
2021-03-04
</td>
<td>
Rebuilding index
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jennysjaarda/PSYMETAB/blob/941b66dff7d9f6ebb58759cb9e2ab4a610d9aab3/analysis/index.Rmd" target="_blank">941b66d</a>
</td>
<td>
Jenny Sjaarda
</td>
<td>
2021-03-02
</td>
<td>
add new Rmd files and respective html files
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jennysjaarda/PSYMETAB/blob/85955b30c2e66a5d9a2383739e037050dbc634a5/analysis/index.Rmd" target="_blank">85955b3</a>
</td>
<td>
Jenny Sjaarda
</td>
<td>
2020-06-23
</td>
<td>
Update site
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jennysjaarda/PSYMETAB/85955b30c2e66a5d9a2383739e037050dbc634a5/docs/index.html" target="_blank">85955b3</a>
</td>
<td>
Jenny Sjaarda
</td>
<td>
2020-06-23
</td>
<td>
Update site
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jennysjaarda/PSYMETAB/46477dd771ba63ffa5d44a0d4f53a1a835a71c87/docs/index.html" target="_blank">46477dd</a>
</td>
<td>
Jenny Sjaarda
</td>
<td>
2019-12-06
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jennysjaarda/PSYMETAB/blob/b503ef05e45518dbd64f6b84c3c6f4e0d0fd64f8/analysis/index.Rmd" target="_blank">b503ef0</a>
</td>
<td>
Sjaarda Jennifer Lynn
</td>
<td>
2019-12-06
</td>
<td>
add more details to website
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jennysjaarda/PSYMETAB/d1e539cbc164ef86728db40b92cd5d275ac36a66/docs/index.html" target="_blank">d1e539c</a>
</td>
<td>
Jenny Sjaarda
</td>
<td>
2019-12-06
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jennysjaarda/PSYMETAB/blob/487b5f56d9bdfd9bc979405b72e23980daf9dd69/analysis/index.Rmd" target="_blank">487b5f5</a>
</td>
<td>
Sjaarda Jennifer Lynn
</td>
<td>
2019-12-06
</td>
<td>
update website, add qc description
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jennysjaarda/PSYMETAB/9f1ba5e2539136e5a4cdcbba996ba481e11d4cc0/docs/index.html" target="_blank">9f1ba5e</a>
</td>
<td>
Jenny Sjaarda
</td>
<td>
2019-12-06
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jennysjaarda/PSYMETAB/blob/d480e3512e5ae9fdcbc8b17e644075b95a76a502/analysis/index.Rmd" target="_blank">d480e35</a>
</td>
<td>
Jenny
</td>
<td>
2019-12-04
</td>
<td>
misc annotations
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jennysjaarda/PSYMETAB/12223b34fc502251f187d1583fbc0b5761aa9f65/docs/index.html" target="_blank">12223b3</a>
</td>
<td>
Jenny Sjaarda
</td>
<td>
2019-12-02
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jennysjaarda/PSYMETAB/blob/dafe34662403daf678c932d4681bb4a92df93bf0/analysis/index.Rmd" target="_blank">dafe346</a>
</td>
<td>
Jenny
</td>
<td>
2019-12-02
</td>
<td>
modify index
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jennysjaarda/PSYMETAB/blob/0dd02a792ccdb998d5650da4f6b713f0d4a46c83/analysis/index.Rmd" target="_blank">0dd02a7</a>
</td>
<td>
Jenny
</td>
<td>
2019-12-02
</td>
<td>
modify website
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jennysjaarda/PSYMETAB/2849dcbac2c41fe9110cee42ccf040d373821b69/docs/index.html" target="_blank">2849dcb</a>
</td>
<td>
Jenny Sjaarda
</td>
<td>
2019-12-02
</td>
<td>
wflow_git_commit(all = T)
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jennysjaarda/PSYMETAB/blob/3d512c23c59e94e41cec64db5916f13532bfb5fd/analysis/index.Rmd" target="_blank">3d512c2</a>
</td>
<td>
Sjaarda Jennifer Lynn
</td>
<td>
2019-12-02
</td>
<td>
modify website and drake plans
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jennysjaarda/PSYMETAB/blob/8e743a865994963ce285c88620bfdb37798b062d/analysis/index.Rmd" target="_blank">8e743a8</a>
</td>
<td>
Sjaarda Jennifer Lynn
</td>
<td>
2019-11-27
</td>
<td>
drakeplan
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jennysjaarda/PSYMETAB/97b2b48ab966ae123bccd64aacf2e0d331952d2e/docs/index.html" target="_blank">97b2b48</a>
</td>
<td>
Jenny
</td>
<td>
2019-11-26
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jennysjaarda/PSYMETAB/blob/2db9bcfcc7febefb29ff93ea2a30e41c0f0156f2/analysis/index.Rmd" target="_blank">2db9bcf</a>
</td>
<td>
Jenny
</td>
<td>
2019-11-26
</td>
<td>
wflow_publish(all = T)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jennysjaarda/PSYMETAB/2db9bcfcc7febefb29ff93ea2a30e41c0f0156f2/docs/index.html" target="_blank">2db9bcf</a>
</td>
<td>
Jenny
</td>
<td>
2019-11-26
</td>
<td>
wflow_publish(all = T)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jennysjaarda/PSYMETAB/5fb939e5f8209480d44c3c98625a40ff85c59e44/docs/index.html" target="_blank">5fb939e</a>
</td>
<td>
Jenny Sjaarda
</td>
<td>
2019-11-26
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jennysjaarda/PSYMETAB/blob/7bae873483c621107e61ecd6592546ed1aef112a/analysis/index.Rmd" target="_blank">7bae873</a>
</td>
<td>
Jenny Sjaarda
</td>
<td>
2019-11-26
</td>
<td>
Start workflowr project.
</td>
</tr>
</tbody>
</table>
</div>

<hr>
</div>
</div>
</div>







<p><a href="https://github.com/jennysjaarda/PSYMETAB"><img src="https://img.shields.io/badge/last%20change-2021--05--31-brightgreen.svg" alt="Last-changedate" /></a> <a href="https://cran.r-project.org/"><img src="https://img.shields.io/badge/R%3E%3D-3.5.0-brightgreen.svg" alt="minimal R version" /></a></p>
<p>This website contains all information and results about this research project.</p>
<p>Project details and overview can be found in the <a href="about.html">About</a> page.</p>
<p>See <a href="setup.html">Setup</a> for details on setting up the project and the various analyses for this project.</p>
<p>See <a href="plans.html">Drake plans</a> for a visual representation of analytical pipelines.</p>
<div id="useful-shortcuts" class="section level2">
<h2>Useful shortcuts</h2>
<ol style="list-style-type: decimal">
<li><a href="data_sources.html">Data sources</a> for details on data sources and origins.</li>
<li><a href="data_processing_in_genomestudio.html">Overview of processing in genomestudio</a>.</li>
<li><a href="genetic_quality_control.html">Genetic quality control</a>.</li>
<li><a href="pheno_quality_control.html">Phenotypic quality control</a>.</li>
<li><a href="extract_genetic_data.html">Genetic data extraction protocol</a>.</li>
<li><a href="GWAS_method.html">GWAS methods</a>.</li>
<li><a href="GWAS_results.html">GWAS results</a>.</li>
</ol>
<p>Powered by <a href="https://github.com/ropensci/drake">drake</a> + <a href="https://github.com/jdblischak/workflowr">workflowr</a></p>
</div>
<div id="useful-papers" class="section level2">
<h2>Useful papers</h2>
<ul>
<li><a href="https://pubmed.ncbi.nlm.nih.gov/16314759/">Clozapine-induced weight gain: a study in monozygotic twins and same-sex sib pairs.</a></li>
<li><a href="https://pubmed.ncbi.nlm.nih.gov/32126890/">A new genetic locus for antipsychotic-induced weight gain: A genome-wide study of first-episode psychosis patients using amisulpride (from the OPTiMiSE cohort).</a></li>
<li><a href="https://pubmed.ncbi.nlm.nih.gov/22316639/">Psychotropic drug-induced weight gain and other metabolic complications in a Swiss psychiatric population.</a></li>
<li><a href="https://www.nature.com/articles/mp2011109">Pharmacogenetics of antipsychotic-induced weight gain: review and clinical implications.</a></li>
</ul>
</div>
<div id="current-analyses" class="section level2">
<h2>Current Analyses</h2>
<div id="gwas-in-psymetab-our-study" class="section level3">
<h3>GWAS in PSYMETAB (our study)</h3>
<ul>
<li>Run within various subsets.</li>
<li>Subsets were defined based on drug use, for various drug groups of interest (we had 9 groups of drugs that we analyzed).</li>
<li>Phenotypes included various measures of the evolution of BMI: BMI change, BMI slope, weighted slope for SE, etc.</li>
<li>Restricted to only European subset.</li>
<li>Phenotypes were residualized for: age, age^2, sex, PC1-20, follow-up time, follow-up time^2, and baseline BMI (Zoltan: we discussed this last time, but to again clarify - should follow-up time be included, it seems redundant if we are analyzing slope which already considers time between BMI measures).</li>
</ul>
</div>
<div id="gwas-in-ukbb" class="section level3">
<h3>GWAS in UKBB</h3>
<ul>
<li>Phenotype: BMI slope.</li>
<li>GWAS restricted to participants having at least 2 BMI measures restricted to unrelated, British individuals.</li>
<li>Phenotype was first residualized for: age, sex, PC1-40, and baseline BMI</li>
</ul>
</div>
<div id="comparison-between-psymetab-and-ukbb" class="section level3">
<h3>COMPARISON BETWEEN PSYMETAB and UKBB:</h3>
<ul>
<li>In PSYMETAB, I took only endpoints involving slope since this would represent the best comparison to UKBB.</li>
<li>I extracted nominally significant hits in PSYMETAB (p &lt; 5e-06) (as there were very few less with p&lt; 5e-08).</li>
<li>For these SNPs, I compared with UKBB using a standard heterogeneity statistic.</li>
<li>Limited to only GW-sig results based on het p-value.</li>
<li>Identified 68 GW-significant SNPs (i.e. SNPs that show significant difference between some PSYMETAB subset and UKBB based on a BMI slope measure, for only those SNPs in PSYMETAB that have p &lt; 5e-06).</li>
</ul>
<ol style="list-style-type: decimal">
<li>Test top SNPs from our GWAS (within drug subgroups) with BMI in UKBB within a group defined as under the same drug as the association in our psychiatric cohort (using drug information data in UKBB). Note that in contrast to our psychiatric cohort, we don’t have information for how long a UKBB participant has been taking a specific drug.</li>
<li>To justify (1), ensure that BMI is a good proxy for change in BMI (i.e. the phenotype used in the psychiatric cohort), correlate BMI at end of follow-up with BMI slope in our psychiatric cohort.</li>
<li>For rare (0.01 &lt; MAF &lt; 0.05) SNPs, perform permutation test (i.e. shuffle phenotype ~1M times and perform same association: pheno ~ SNP + covars) to see if the effect found with the real data (i.e. unshuffled phenotype) is significantly larger than using the 1M shuffled phenotypes and not just some artifact of low AF.</li>
<li>Meta-analyze GWAS results across drug subgroups and use Cochran’s Q test (i.e. test for heterogeneity) to determine effects differ across subgroups.</li>
</ol>
</div>
</div>
<div id="to-do" class="section level2">
<h2>TO DO</h2>
<p><em>Analyses on going before mat-leave:</em></p>
<ol start="0" style="list-style-type: decimal">
<li><p>Run GWAS in UKBB with BMI change for comparison to GWAS in PSYMETAB. <strong>This analysis was completed before maternity leave, but noticed that the GWAS (using BGENIE with Ninon and Ele’s script) excluded HRC SNPs and should be performed on the complete SNP list. This analysis was running before maternity leave and needs to be checked that is ran properly.</strong> <em>See <code>launch_bgenie</code> function</em></p></li>
<li><p>Test top SNPs from our GWAS (within drug subgroups) with BMI in UKBB within a group defined as under the same drug as the association in our psychiatric cohort (using drug information data in UKBB). Note that in contrast to our psychiatric cohort, we don’t have information for how long a UKBB participant has been taking a specific drug.</p></li>
<li><p>To justify (1), ensure that BMI is a good proxy for change in BMI (i.e. the phenotype used in the psychiatric cohort), correlate BMI at end of follow-up with BMI slope in our psychiatric cohort.</p></li>
<li><p>For rare (0.01 &lt; MAF &lt; 0.05) SNPs, perform permutation test (i.e. shuffle phenotype ~1M times and perform same association: pheno ~ SNP + covars) to see if the effect found with the real data (i.e. unshuffled phenotype) is significantly larger than using the 1M shuffled phenotypes and not just some artifact of low AF.</p></li>
<li><p>Meta-analyze GWAS results across drug subgroups and use Cochran’s Q test (i.e. test for heterogeneity) to determine effects differ across subgroups.</p></li>
<li><p>Added <code>logtotal_methyl_caffeine</code> to PRS analysis for Nermine. <em>See <code>prs_analysis</code> plan</em></p></li>
</ol>
</div>
<div id="acknowledgements" class="section level2">
<h2>Acknowledgements</h2>
<p>Primary analysis performed by: <a href="jennysjaarda@gmail.com">Jenny Sjaarda</a>.<br />Project supervisor: <a href="https://www.unil.ch/fbm/fr/home/menuinst/la-releve-academique/nominations--promotions/professeurs-a-a-z/d-g/eap-chin-bin.html">Professor Chin Bin Eap</a>.<br />Analysis support: <a href="https://wp.unil.ch/sgg/people/zoltan/">Professor Zoltan Kutalik</a>.</p>
<p>This work has been supported by the <a href="http://www.snf.ch/en/funding/Pages/default.aspx">Swiss National Science Foundation</a>.</p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo"
  data-toggle="collapse" data-target="#workflowr-sessioninfo"
  style = "display: block;">
  
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>

<div id="workflowr-sessioninfo" class="collapse">

<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.5.3 (2019-03-11)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

Matrix products: default
BLAS: /data/sgg2/jenny/bin/R-3.5.3/lib64/R/lib/libRblas.so
LAPACK: /data/sgg2/jenny/bin/R-3.5.3/lib64/R/lib/libRlapack.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] workflowr_1.6.0

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.3      rprojroot_1.3-2 packrat_0.5.0   digest_0.6.25  
 [5] later_1.0.0     R6_2.4.1        backports_1.1.6 git2r_0.26.1   
 [9] magrittr_1.5    evaluate_0.14   stringi_1.4.5   rlang_0.4.5    
[13] fs_1.3.1        promises_1.1.0  whisker_0.4     rmarkdown_1.18 
[17] tools_3.5.3     stringr_1.4.0   glue_1.4.0      yaml_2.2.0     
[21] httpuv_1.5.2    xfun_0.11       compiler_3.5.3  htmltools_0.4.0
[25] knitr_1.26     </code></pre>
</div>


</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>



</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
