---
title: "Extracting genetic data"
site: workflowr::wflow_site
output:
  workflowr::wflow_html:
    toc: true
---

**The following document outlines the procedure to extract genetic data located on the `HPC1` servers.**


- Text file.
- One line per SNP.
- Name of file is not important, but should have some simple description of the SNP list (`${snp_list}`).

# 2. Transfer input file to HPC1 server.

- Connect to `HPC1` server via filezilla:
  - host: `hpc1.chuv.ch`
  - username: `je4649`
  - password: `XXXX`
  - port: `22`
- Navigate to `/data/sgg2/jenny/projects/PSYMETAB/data/raw/extractions`
- If necessary, create directory for person who made data extraction request (`requestor`) with the format: `FirstnameLastname`.
- Save text file from #1 into the folder: `/data/sgg2/jenny/projects/PSYMETAB/data/raw/extractions/${requestor}` (Note: to maintain file organization, please maintain only *one* folder per requestor).

# 3. Create and run extraction script for "requestor".

- Login to `HPC1` using PUTTY:
  - hostname: `je4649@hpc1.chuv.ch`
  - password: `XXXX`
- Launch sbatch master script from master as follows: `sbatch $projects/PSYMETAB/code/extractions/master_extract.sh` (this assumes projects is defined as `/data/sgg2/jenny/projects`, if not replace `$projects` with explicit path or define projects before hand, if logged in as `je4649`, `project` is automatically defined).
- Code details:
  - An extraction script template is found here: `/data/sgg2/jenny/projects/PSYMETAB/code/extractions/template_extract.sh`.
  - The master script (`/data/sgg2/jenny/projects/PSYMETAB/code/extractions/master_extract.sh`) will automatically do the following:
    - Create a new extraction script for each requestor saved to the name: `/data/sgg2/jenny/projects/PSYMETAB/code/extractions/${requestor}.sh`.
    - Replace all instances of  `FirstnameLastname` in Template with new "requestor" (e.g. replace **Aurelie** with **FirstnameLastname**), there should be four replace instances: (1) SBATCH `job-name`, (2) folder `input`, (3) folder `ouput`, (4) comment at end of file indicating how to launch the script.
    - Run the extraction script for each requestor as an `sbatch` job.
  - The extraction script will only try to perform the extraction for a given file (i.e. the memory intensive portion of the script) if a folder does not already exists which corresponds to that SNP lists here: `/data/sgg2/jenny/projects/PSYMETAB/data/processed/extractions/${resquester}/${snp_list}`

# 4. Summary of result files.

- Results will be saved to the following location: `/data/sgg2/jenny/projects/PSYMETAB/data/processed/extractions/${resquester}/${snp_list}`.
- `${snp_list}_rsids.txt`: list of rsIDs with empty characters removed (in most cases this should be identical to the input file from #1).
- `${snp_list}_SNP_info.txt`: info file for each rsID, extracted from HRC list (see [HRC website](http://www.haplotype-reference-consortium.org/home)), columns correspond to: `#CHROM	POS	ID	REF	ALT	AC	AN	AF	AC_EXCLUDING_1000G	AN_EXCLUDING_1000G	AF_EXCLUDING_1000G	AA`.
- For each chromosome (`$chr`), extracted from imputed data, there are the following files from extracting the chromosome specific genetic data using [PLINK](https://www.cog-genomics.org/plink/2.0/):
  - `${snp_list}_chr${chr}.txt`: list of rsIDs on `chr${chr}`
  - `${snp_list}_chr${chr}.psam`
  - `${snp_list}_chr${chr}.pvar`
  - `${snp_list}_chr${chr}.pgen`
  - `${snp_list}_chr${chr}.pgen`
  - `${snp_list}_chr${chr}.raw`
  - `${snp_list}_chr${chr}.log`
- `${snp_list}_extraction.txt`: merged genetic data from imputed files.
- Some SNPs of interest (usually insertions and deletions) were genotyped directly on  our custom SNP panel, but were not imputed (HRC reference panel does not include insertion/deletion SNPs) and therefore would not be extracted and included in the above extraction. If the "requestor" has requested such SNPs, there are also the following files provided:
  - `${snp_list}_missing_snps.txt`: list of SNPs that were not extracted from imputed files (if this file is empty, all SNPs were in imputed data set).
  - `${snp_list}_missing_snps_extract.fam`: PLINK extraction from original genotype data.
  - `${snp_list}_missing_snps_extract.bim`: PLINK extraction from original genotype data.
  - `${snp_list}_missing_snps_extract.bed`: PLINK extraction from original genotype data.
  - `${snp_list}_missing_snps_extract.raw`: PLINK extraction from original genotype data.
  - `${snp_list}_missing_snps_extract.log`: PLINK extraction from original genotype data.
  - `${snp_list}_extraction_geno.txt`: merged genetic data, same file as above, but with additional SNPs extracted from original genotype data, this is the **main results file**:
    - First column is `GPCR` ID.
    - Subsequent **N** columns correspond **N** SNPs extracted from imputed data.
    - One column for ethnicity (one of: "CEU" (European), "EA" (East Asian), "MIXED", "NA" (Native American), "YRI" (African))
    - Following 100 columns correspond to genetic principal components.
    - *Note*: it is **strongly** encouraged to only us the `CEU` samples as this our largest and cleanest sample. If using only European, adjust for at least the first 10 PCs in your analysis. If you need to include additional ethnic groups, you have two options: (1) analyze each ethnic group seperately, adjusting for PCs **or** (2) analyze the entire group together without PCs (PCs were calculated in each ethnic group **seperately** so you cannot include them in a model where all ethnic groups are included).
  - `${snp_list}_missing_geno_snps.txt`: list of SNPs that are still missing after extracting SNPs from original genotyped data.
    - *Note*: data from the original genotype data has not been cleaned or filtered, so should be used with some caution.
    - List of SNPs extracted from original genotype data can be seen in the bim file: `${snp_list}_missing_snps_extract.bim`.
- All data from imputed files has been processed according to the description outlined in the [Genotype quality control page](genetic_quality_control.html) (i.e. SNPs filtered for MAF, HW, info, etc. and participants filtered for sex check, relatedness, etc.).

# 5. Send results to "requestor".

- The simplest way is to likely to move the main results file (`${snp_list}_extraction_geno.txt`) to a CHUV computer via filezilla.
- Results then need to be sent to Claire or Celine to convert the GPCR ID to a patient ID.
- Celine and/or Claire will then transfer the recoded data to the `requestor`.
