---
title: "GWAS Method"
site: workflowr::wflow_site
output:
  workflowr::wflow_html:
    toc: true
---

```{css include=FALSE}
pre code, pre, code {
  white-space: pre !important;
  overflow-x: scroll !important;
  word-break: keep-all !important;
  word-wrap: initial !important;
}
```

```{r setup, include = FALSE}
options(scipen=999)
source("R/packages.R")
source("R/functions.R")
source("R/settings.R")
```


```{r load_data, include = FALSE}
loadd(pheno_raw)
loadd(pheno_baseline)
loadd(pheno_followup)
loadd(GWAS_input)
loadd(baseline_gwas_info)
loadd(interaction_gwas_info)
loadd(subgroup_gwas_info)

pheno_baseline_dup <- pheno_baseline
pheno_followup_dup <- pheno_followup
```

```{r drake_functions, include = FALSE}


count_6mo_NAs <-
  function(pheno_baseline,
           test_drugs,
           test_drugs_num,
           low_inducers,
           high_inducers) {
    return_list <- list()
    out <- numeric()
    drug_class <-
      unlist(test_drugs %>% dplyr::select(class) %>% dplyr::slice(test_drugs_num))
    drug_list <-
      unlist(test_drugs %>% dplyr::select(drugs) %>% dplyr::slice(test_drugs_num))
    for (pheno in interaction_vars) {
      for (j in 1:dim(pheno_baseline)[1]) {
        result <- pheno_baseline %>% slice(j)
        
        
        x  <-
          classify_drugs(result, pheno, drug_list, low_inducers, high_inducers)
        drug_info <- clean_drugs(result, pheno)
        drugs_taken <-
          result %>% dplyr::select(starts_with("AP1_Drug_")) %>% unlist(., use.names =
                                                                          F)
        drugs_taken <- drugs_taken[!is.na(drugs_taken)]
        if (is.na(x$pheno_change_6mo) &
            !(is.na(x$pheno_diff)) &
            length(drugs_taken) > 1 &
            !all(is.na(drug_info$pheno_change_6mo))) {
          out <- c(out, j)
        }
        
        
      }
      
      return_list[[paste0(drug_class, "_", pheno)]] <- out
    }
    
    return(return_list)
    
  }



example_df <- function() {
  Participant <- c(rep('A', 10), rep('B', 5), rep('C', 8))
  Drug <- c(rep('XX', 3), rep('XY', 3), rep('XZ', 2), rep('XX', 2),
            rep('YX', 3), rep('XY', 2),
            rep('XX', 3), rep('YY', 4), rep('YZ', 1))
  Date <- as.Date(c('2010-1-1', '2010-3-1', '2010-10-1', '2015-1-1', '2010-5-1', '2017-1-1', '2017-2-1', '2017-4-1', '2017-5-1', '2018-6-1',
                         '2010-1-1', '2010-3-1', '2010-12-1', '2011-3-1', '2011-4-1',
                         '2010-1-1', '2010-2-1', '2015-1-1', '2015-1-1', '2015-2-1', '2015-6-1', '2016-5-1', '2019-1-1'))
  Follow_up <- c(rep(10001, 3), rep(10002, 3), rep(10003, 2), rep(10004, 2),
                 rep(10005, 3), rep(10006, 2),
                 rep(10007, 2), rep(10008, 1), rep(10009, 4), rep(10010, 1))
  BMI <- rnorm(length(Participant), 26, sd=2)
  
  temp <- tibble(Participant=Participant, Drug=Drug, Date=Date, Follow_up=Follow_up, BMI=BMI)
  
  out <- temp %>% group_by(Participant) %>% arrange(Follow_up, Date) %>%
    mutate(Follow_up_mod = rename_meds(Drug, Follow_up, Date)) %>%
    group_by(Participant,Follow_up_mod) %>%
    mutate(Follow_up_duration= as.numeric(max(Date)-min(Date)))%>% mutate(Follow_up_duration = na_if(Follow_up_duration, 0)) %>%
    mutate(Num_followups = n()) %>%
    mutate(date_difference_first =  as.numeric(difftime(Date, first(Date)), units = "days")) %>%
    mutate(BMI_change = X_diff(BMI)) 
  return(out)
  
}


count_outliers <- function(pheno_baseline, pheno_followup, caffeine_vars, high_inducers, med_inducers, low_inducers){
  na_to_none <- function(x) ifelse(is.na(x),'NONE',x)
  linear_pheno <- pheno_baseline %>% dplyr::select(matches('^FID$|^IID$|_start_Drug_1'), paste0(caffeine_vars, "_caffeine"))
  linear_covar <- pheno_baseline %>% dplyr::select(matches('^FID$|^IID$|^sex$|^Age_Drug_1$|Age_sq_Drug_1$|^PC[0-9]$|^PC[1][0-9]$|^PC20|^Age_caffeine'))

  list_covar <- list(linear=linear_covar)
  list_pheno <- list(linear=linear_pheno)
  out <- numeric()
  
  for(sub in names(pheno_followup))
  {
    data_drug <- pheno_followup[[sub]]
    pheno <-  unlist(str_split(sub, "[.]"))[2]
    drug <-  unlist(str_split(sub, "[.]"))[1]
    drugs_analyzed <- test_drugs[which(test_drugs$class==drug),] %>% pull(drugs) %>% unlist()
    remove_drugs <- numeric()
    count <- 1
    for(drug_list_name in c("high_inducers", "med_inducers", "low_inducers")){
      drug_list <- get(drug_list_name)
      if(!any(drugs_analyzed %in% drug_list)){
        remove_drug_i <- drug_list
        remove_drugs <- c(remove_drugs, remove_drug_i)
      }
      if(any(drugs_analyzed %in% drug_list)){
        break
        #remove_drug_i <- drug_list[which(!drug_list %in% drugs_analyzed)]
        #remove_drugs <- c(remove_drugs, remove_drug_i)
      }
      count <- count + 1
    }

    interaction_outcome_pheno <- interaction_outcome[which(grepl(pheno, interaction_outcome))]
    
    data_drug_sensitivity <- data_drug %>% mutate(high_inducer=replace(high_inducer, high_inducer_drug_name %in% remove_drugs, NA)) %>%
      mutate_at(interaction_outcome_pheno, ~ifelse(is.na(high_inducer), NA, .)) %>%
      dplyr::select(FID, IID, high_inducer, interaction_outcome_pheno) %>% rename_at(vars(-FID,-IID),function(x) paste0(x,"_sensitivity"))

    all_var <- data_drug %>% dplyr::select(matches('^FID$|^IID$|^bmi_start$|^high_inducer$|^age_|^follow_up_time',ignore.case = F), paste0(pheno, "_start"), paste0(pheno, outcomes)) %>% #interaction_outcome) %>%
      left_join(data_drug_sensitivity) %>%
      mutate_at(c("high_inducer", "high_inducer_sensitivity"), as.character) %>%
      mutate_at(vars(high_inducer, high_inducer_sensitivity),  ~(recode(.,"0" = "NoDrug", "1" = "Drug"))) %>%
      rename_at(vars(-FID,-IID),function(x) paste0(x,"_",drug)) %>%
      mutate_if(.predicate = is.character,.funs = na_to_none)
     

    outliers_removed <- all_var %>% 
      mutate_at(.vars = c(paste0(pheno, outcomes,"_", drug), paste0(pheno, outcomes,"_sensitivity_", drug)), .funs= ~ifelse(abs(.)>mean(., na.rm=T)+5*sd(., na.rm=T), NA, .))
    
    
    NA_count_w_outliers <- all_var %>%  dplyr::select(c(paste0(pheno, outcomes,"_", drug), paste0(pheno, outcomes,"_sensitivity_", drug))) %>%
      summarise_all(~(sum(is.na(.))))
    
    NA_count_wo_outliers <- outliers_removed %>%  dplyr::select(c(paste0(pheno, outcomes,"_", drug), paste0(pheno, outcomes,"_sensitivity_", drug))) %>%
      summarise_all(~(sum(is.na(.))))
    
    NA_difference <- NA_count_wo_outliers - NA_count_w_outliers
    col_old <- colnames(NA_difference)
    col_new <- gsub(pattern = paste0(pheno, "_"),replacement = "pheno_", x  = col_old)
    col_new <- gsub(pattern = paste0("_", drug),replacement = "", x  = col_new)
    
    NA_difference <- cbind(sub, NA_difference)
    colnames(NA_difference) <- c("drug_cat", col_new)
    out <- rbind(out, NA_difference)
  }

  return(out)
}


```


```{r drake_plans, include = FALSE}


GWAS_process <- drake_plan(
  test_drugs_num = 1:dim(!!test_drugs)[1],
  NA_6mo = target(count_6mo_NAs(!!pheno_baseline_dup, !!test_drugs, test_drugs_num, !!low_inducers, !!high_inducers),
                  dynamic = map(test_drugs_num)),
  example_data = example_df(),
  num_outliers_removed = count_outliers (!!pheno_baseline_dup, !!pheno_followup_dup, !!caffeine_vars, test_drugs[1], !!high_inducers, !!med_inducers, !!low_inducers)
  
)

# make(GWAS_process)

num_outliers_removed <- count_outliers (pheno_baseline_dup, pheno_followup_dup, caffeine_vars, high_inducers, med_inducers, low_inducers)
  
loadd(NA_6mo)
loadd(example_data)
```


**The following document outlines and summarizes the method used for performing the GWAS analyses performed in PSYMETAB, for an overview of the results see [GWAS Results](GWAS_results.html).**

# Sample data. 

Original data has the following structure: 

```{r sample_data, echo = FALSE}

example_data %>% dplyr::filter(Participant == "A") %>% kable() %>%
  kable_styling(full_width = F, position = "left") %>% scroll_box(width = "100%")
example_data %>% dplyr::filter(Participant == "B") %>% kable() %>%
  kable_styling(full_width = F, position = "left") %>% scroll_box(width = "100%")
example_data %>% dplyr::filter(Participant == "C") %>% kable() %>%
  kable_styling(full_width = F, position = "left") %>% scroll_box(width = "100%") 

```

Need to transform the above data to have one row per participant with the following columns:

- Drug (Yes/No),
- Possible outcomes of interest (BMI change, slope, weighted slope, etc).
- Relevant covariates (age, sex, follow-up time, etc).

Questions to consider:

- If we are interested in drug  `XX`, which follow-up do we choose (`10001` or `10004`)? 
- If we are interested in a drug (e.g. `ZZ`) which is not taken by our patient, which follow-up do we use as control?


# Data filtering and quality control. 

Follow-ups (i.e. single rows as in sample data above) were removed if:

- Missing height data. 
- Missing weight data. 
- Height +/- 5SD outside mean (this removed patients with the following heights (cm): `106 106 106 106 106 106 106 106 116  96  96  90`).
- Missing date. 
- Missing drug information (column name `AP1`).
- Multiple sexes listed for the same participant (i.e. two different follow-up listed both M and F).
- If multiple follow-ups for the same drug, consider to only round 1. 

Data corrections: 

- Height was defined as mean of all heights. 
- `Risperdal` was redefined as `Risperidone`. 
- `Paliperidone` was redefined as `Risperidone`. 
- Drugs with `depot` or `retard` appendix were considered as original drug. 
- All values equal to `999` were set to missing. 
- If any sex field was missing, sex was taken from other follow-ups (if possible).

# General info. 

- Analyses were performed in each ethnicity seperately and then meta-anlayzed.
- Min follow-up time was set as `r min_follow_up` days (i.e. follow-up <= `r min_follow_up` days).
- Some analyses were restricted to ~6, 3 or 1 month(s), of followup (i.e. follow-up <= `r follow_up_6mo`,  `r follow_up_3mo`, `r follow_up_1mo` days, respectively).

# Drug classification. 

After discussion with others, it was decided to define drugs into three classes based on their propensity to induce weight gain:

1. **Low**: AP1 pour faible risque de prise de poids:

Amisulpride, Aripiprazole, Brexpiprazole, Cariprazine, Carbamazepine, Chlorprothixene, Flupentixol, Fluphenazine, Haloperidol, Lurasidone,
Pipampérone, Promazine, Sertindole, Sulpiride, Tiapride.

2. **Medium**: AP2 pour risque intermédiaire de prise de poids:

Amitriptyline, Asenapine, Clomipramine, Dibenzepine, Doxepine, Imipramine, Lévomépromazine, Lithium, Mirtazapine,
Nortriptyline, Opipramol, Palipéridone, Quétiapine, Rispéridone, Trimipramine,  Zuclopenthixol.

3. **High**: AP3 pour risque élevé de prise de poids:

Clozapine, Olanzapine, Valproate.

Specifically, high, medium and low inducers were defined as follows: 

```{r inducer_class, echo = TRUE}
high_inducers
med_inducers
low_inducers
```

An additional group of drugs were defined as `antipsychotics` which excludes mood stabilizers and antidepressants as follows (*as of 25/09/2020, no analyses have been performed with this group*):
```{r antipsychotic_definition, echo = TRUE}
antipsychotics
```

# Drug (case/control) definitions.

- All drugs were defined based on their propensity to induce weight gain, and fall into one of three categories (see above): 
  1. Low inducers 
  2. Medium inducers
  3. High inducers
- To test the SNP*drug effect of a drug (or drug set) of interest participants were classified as either cases or controls. 
- Cases were defined as:
  - Participants who have taken the drug of interest at least once. 
  - Partcipants who have been followed up >= `min_follow_up` (`r min_follow_up` days) for drug(s) of interest.
  - If particpant has multiple follow-ups for the drug of interest that have duration >= `min_follow_up`, the first follow-up is retained.
- Controls were defined as: 
  - Participants who do not meet above criteria. 
  - Only follow-ups with duration > `min_follow_up` are retained.
  - If particpant has multiple alternate follow-ups, *low inducing* drugs are prioritized, followed-by *medium inducing* drugs, and lastly *high inducing* drugs.
  - Note that all controls are followed up for an alternate drug. 
  - Individuals were filtered if they had taken the drug of interest at some point but it was not specifically followed. Therefore the follow-up data set is smaller than the baseline data set. These individuals were removed because we do not know when they took the drug and if it could have had some impact on their weight evolution. 
- In this way, a different set of cases/controls exists for each drug of interest. 

## Drug/drug sets tested. 

The following drug/drug combinations were tested: 

```{r drugs_tested2, echo = FALSE}

test_drugs %>% kable() %>%
  kable_styling(full_width = F, position = "left") 

```

## Sensitivity drug sets.

- A sensitivity data-set was built for each drug set was also tested: 
- Define `max_category` for a set of drugs of interest as `high`, `medium`, or `low` depending on the highest inducing category found in the drug set. 
- Remove all controls which have a follow-up for a drug **higher** than the `max_category`.
- For example, if the gene*quetiapine interaction was tested:
  - `max_category` = `medium`
  - All cases falling in category `high` were removed. 
  - If testing a set of drugs spanning multiple categories (i.e. quetiapine, clozapine, olanzapine), the `max_category` would be `high` and no individuals would be removed in the sensitivity set. 

## Independent drug sets. 

- Based on Zoltan's suggestion, PSYMETAB was divided into subgroups based on different drugs such that each participant is only in one subgroup.
- Because participants can be followed for more than one drug, it is necessary to *select* one drug per participant. 
- Therefore a prioritization method was followed: 
  - If a particpant has been followed for the highest priority drug they are placed in this drug group. 
  - If not, they are checked for the second drug in the prioritization list and either placed in the second drug group or examined to see if they were followed for the next drug in the list. 
  - This process was repeated for all drugs in the prioritization list.
  - If a participant has not been followed for any of the drugs in the prioritization list, they are placed in category `Other`. 
- The following independent drug groups were tested, in the following order. 

```{r drug_prioritization_list, echo = FALSE}

drug_prioritization %>% kable() %>%
  kable_styling(full_width = F, position = "left") 

```

# Phenotype definitions.

- The main phenotypes of interest were built using follow-up data, including:
  - **Pheno change**: raw change between first and last date of follow-up. 
  - **Pheno change 1mo**: raw change between first and last date of follow-up, limited to ~1 months (<= `r follow_up_1mo` days).
  - **Pheno change 3mo**: raw change between first and last date of follow-up, limited to ~3 months (<= `r follow_up_3mo` days).
  - **Pheno change 6mo**: raw change between first and last date of follow-up, limited to ~6 months (<= `r follow_up_6mo` days).
  - **Pheno slope**: beta coefficient of a linear model betwen `BMI` (y) and `days of follow-up` (x) (i.e. `date_difference_first` in example data above).
  - **Pheno slope weight**: weighted beta coefficient from model used for **Pheno slope** by SE (i.e `std_beta <- beta/se`).
  - **Pheno slope 6mo**: same as **Pheno slope** but follow-up time restricted to ~6 months (<= `r follow_up_6mo` days).
  - **Pheno slope weight 6mo**: same as **Pheno slope weight** but follow-up time restricted to ~6 months (<= `r follow_up_6mo` days).
- Phenotypes tested in this way included: 
```{r interaction_vars, echo = TRUE}
interaction_vars
```
- A different follow-up phenotype was defined for each drug combination. 

## Standard (linear) GWAS.

- In addition to the follow-up phenotypes described above, the following baseline variables were tested in a standard GWAS setting:
```{r baseline_vars, echo = TRUE}
baseline_vars 
caffeine_vars
```
- If multiple follow-ups were available for a given individual, the first time-point was used. 
- Each follow-up phenotype (described above) * drug combination was also tested in a linear model including all participants.
  - Note that there would be *very* little difference between some of these phenotypes.
  - Because participants have multiple follow-ups for different drugs, a decision needs to be made when creating a follow-up phenotype, see case/control definitions above.
  - In these crude models, there is no accounting for the drug of interest (this is done in the interaction or subgroup model below) and all individuals are included in the model.
  - For example, the phenotype `BMI change` was defined using all possible case/control definitions. If definied using olanzapine (`olanz`) case/control status, olanzapine follow-ups were used (as cases) where possible, and if not, a low-inducing drug was selected, otherwise the first follow-up is taken (as controls). If defined using olanzapine + clozapine (`olanz_cloz`) case/control status, either olanzapine **or** clozapine follow-ups (whichever came first if a participant had taken both) were selected (as cases) as the follow-up of interest, and if a participant had not been followed for either drug, a low-inducing drug was selected, otherwise the first follow-up is taken (as control). In this way, these two phenotypes would likely be nearly identical. The **only** difference would be for those individuals that had a follow-up other than clozapine which used as the control follow-up in the `olanz` defined phenotype, but had been followed for clozapine at some other point in the dataset. Therefore this individual would have some `drugX` used as the followup period to define the phenotype in the `olonz` analysis, but the `clozapine` follow-up period used in the `olanz_cloz` defined phenotype (in this case `BMI change`). 
- All phenotypes had an inverse variance normal transformation (IVT) applied to ensure a normal distribution before downstream analyses were performed (see function). 
```{r echo_ivt, echo = TRUE}
ivt
```

## Interaction GWAS.

- The following outcomes were tested in a gene*drug interaction GWAS setting
```{r interaction_outcome, echo = TRUE}
interaction_outcome 
```
- As above, all phenotpyes were transformed according to IVT before downstream analyses.  
  - Note that IVT was performed on full data set instead of before sensitivity phenotypes were created.
  - This means that the data used in the sensitivity models is exactly the same as the full models, just with some participants removed.
- Drug definitions were defined as described above.
- According to the drug definition (case/control), there could be participants with:
  - A valid follow-up value if not limited to 6 months.
  - When limited to 6mo, the value is missing (i.e. the first follow-up visit is > `r follow_up_limit` days after the baseline visit), therefore there is no way to define BMI change or slope (as above).
  - Rather than redefine the control value, these participants simply were set to missing. 
  - The number of participants in this criteria for each drug are shown below (in other words, these participants could be redefined with a different follow-up for the 6mo period if we wanted to use a non-NA value):
```{r NA_6mo, echo = FALSE}
sapply(NA_6mo, length)

```  
- Number of models tested = 2 (original/sensitivity model) * length(drugs_tested) * length(interaction_outcome) * 3 ethnic models (European/mixed/meta) = `r 2*dim(test_drugs)[1]*length(interaction_outcome)*3`.
- Values falling outside 5SD of mean were removed, the number removed  for each drug/outcome combination can be seen below: 
```{r NA_6mo_count, echo = FALSE, eval=FALSE}
num_outliers_removed %>% kable() %>%
  kable_styling(full_width = F, position = "left") 
```  

## Subgroup GWAS. 

- For each drug /drug set tested, a GWAS was performed in each group (yes/no on drug(s)).
```{r drugs_tested, echo = FALSE}
test_drugs %>% kable() %>%
  kable_styling(full_width = F, position = "left") 
```
- All outcomes used in interaction (above) were tested as outcomes.
- Number of models tested = 2 (yes/no on drug) * 2 (original/sensitivity model) * length(drugs_tested) * length(interaction_outcome) * 3 ethnic models (European/mixed/meta) = `r 2*2*dim(test_drugs)[1]*length(interaction_outcome)*3`.
- The idea was that a hetergeneity test would be performed to compare the GWAS between the two subgroups to identify associations that differ between the two groups (those under vs those not under a given drug). This would be to identify associations that are specific to a use of a certain drug. However, after further consideration, we decided that this did not make sense as the control group is not an ideal set of controls given that they are also taking some drug, and all have participants some psychiatric condition. 

# Covariate definitions and adjustment.

## Residualization.

- Phenotypes were adjusted for covariates via residualization rather than standard covariate adjustemnt.
- Ethnic groups were residualized seperately since PCs were calculated in each ethnic group seperately and models were subsequently run for each ethnic group. 
- Residualization rather than explicity covariate adjustment allows for far fewer input files into `PLINK` and consequently a much faster run time.

## Standard (Linear) GWAS.

- Baseline/caffeine analyses were adjusted for PC1-20, sex, age, age squared.
- Linear models using follow-up data were adjusted for the same covariates as baseline models plus: 
  - Follow-up time.
  - Follow-up time squared.
  - BMI start.
- If 1mo, 3mo or 6mo variable was used, follow-up time was adjusted accordingly. 
- Variable of interest: **SNP**.
```{r linear_covars, echo = TRUE}
standard_covars 
baseline_covars 
caffeine_covars
baseline_gwas_info %>% kable() %>%
  kable_styling(full_width = F, position = "left") %>% scroll_box(width = "100%", height="500px")
```

## Interaction GWAS.

- Interaction GWAS models were adjusted for PC1-20, sex, plus: 
  - Age started drug (varied depending on drug tested)
  - Age started drug squared.
  - Follow-up time.
  - Follow-up time squared.
  - BMI at start.
- Variable of interest: **SNP*drug_status**.
- To test the interaction variable explicitly:
  - The following model was applied in `PLINK`: _resid_pheno ~ SNP + drug_status + SNP*drug_status_
  - Therefore, the phenotype of interest was residualized for all variables above, but the interaction term and it's parts were included explicitly as covariates.
- Detailed covariate adjustment for each phenotype is shown below:  
```{r interaction_covars, echo = TRUE}
standard_covars 
interaction_gwas_info %>% kable() %>%
  kable_styling(full_width = F, position = "left") %>% scroll_box(width = "100%", height="500px")
```

## Subgroup GWAS. 

- Same adjustments as interaction GWAS but without drug status (since models are already split by drug status) and interaction term.
- For subgroup models, residualization was performed in each ethnic group sepeartely (as in models above) but also in each drug category seperately.
- In other words, phenotypes were residualized in each stratum (i.e. drug case/control subgroups) seperately in case there is any interaction between stratum and covariate. 
- Variable of interest: **SNP**.
