---
title: "GWAS"
site: workflowr::wflow_site
output:
  workflowr::wflow_html:
    toc: true
---

```{css include=FALSE}
pre code, pre, code {
  white-space: pre !important;
  overflow-x: scroll !important;
  word-break: keep-all !important;
  word-wrap: initial !important;
}
```

```{r setup, include = FALSE}
options(scipen=999)
source("code/packages.R")
source("code/functions.R")
source("code/settings.R")
```


```{r load_data, include = FALSE}
loadd(pheno_raw)
loadd(pheno_baseline)
loadd(pheno_followup)
loadd(GWAS_input)
loadd(baseline_gwas_info)
loadd(interaction_gwas_info)
loadd(subgroup_gwas_info)

pheno_baseline_dup <- pheno_baseline
pheno_followup_dup <- pheno_followup
```

```{r drake_functions, include = FALSE}


count_6mo_NAs <-
  function(pheno_baseline,
           test_drugs,
           test_drugs_num,
           low_inducers,
           high_inducers) {
    return_list <- list()
    out <- numeric()
    drug_class <-
      unlist(test_drugs %>% dplyr::select(class) %>% dplyr::slice(test_drugs_num))
    drug_list <-
      unlist(test_drugs %>% dplyr::select(drugs) %>% dplyr::slice(test_drugs_num))
    for (pheno in interaction_vars) {
      for (j in 1:dim(pheno_baseline)[1]) {
        result <- pheno_baseline %>% slice(j)
        
        
        x  <-
          classify_drugs(result, pheno, drug_list, low_inducers, high_inducers)
        drug_info <- clean_drugs(result, pheno)
        drugs_taken <-
          result %>% dplyr::select(starts_with("AP1_Drug_")) %>% unlist(., use.names =
                                                                          F)
        drugs_taken <- drugs_taken[!is.na(drugs_taken)]
        if (is.na(x$pheno_change_6mo) &
            !(is.na(x$pheno_diff)) &
            length(drugs_taken) > 1 &
            !all(is.na(drug_info$pheno_change_6mo))) {
          out <- c(out, j)
        }
        
        
      }
      
      return_list[[paste0(drug_class, "_", pheno)]] <- out
    }
    
    return(return_list)
    
  }



example_df <- function() {
  Participant <- c(rep('A', 10), rep('B', 5), rep('C', 8))
  Drug <- c(rep('XX', 3), rep('XY', 3), rep('XZ', 2), rep('XX', 2),
            rep('YX', 3), rep('XY', 2),
            rep('XX', 3), rep('YY', 4), rep('YZ', 1))
  Date <- as.Date(c('2010-1-1', '2010-3-1', '2010-10-1', '2015-1-1', '2010-5-1', '2017-1-1', '2017-2-1', '2017-4-1', '2017-5-1', '2018-6-1',
                         '2010-1-1', '2010-3-1', '2010-12-1', '2011-3-1', '2011-4-1',
                         '2010-1-1', '2010-2-1', '2015-1-1', '2015-1-1', '2015-2-1', '2015-6-1', '2016-5-1', '2019-1-1'))
  Follow_up <- c(rep(10001, 3), rep(10002, 3), rep(10003, 2), rep(10004, 2),
                 rep(10005, 3), rep(10006, 2),
                 rep(10007, 2), rep(10008, 1), rep(10009, 4), rep(10010, 1))
  BMI <- rnorm(length(Participant), 26, sd=2)
  
  temp <- tibble(Participant=Participant, Drug=Drug, Date=Date, Follow_up=Follow_up, BMI=BMI)
  
  out <- temp %>% group_by(Participant) %>% arrange(Follow_up, Date) %>%
    mutate(Follow_up_mod = rename_meds(Drug, Follow_up, Date)) %>%
    group_by(Participant,Follow_up_mod) %>%
    mutate(Follow_up_duration= as.numeric(max(Date)-min(Date)))%>% mutate(Follow_up_duration = na_if(Follow_up_duration, 0)) %>%
    mutate(Num_followups = n()) %>%
    mutate(date_difference_first =  as.numeric(difftime(Date, first(Date)), units = "days")) %>%
    mutate(BMI_change = X_diff(BMI)) 
  return(out)
  
}


count_outliers <- function(pheno_baseline, pheno_followup, caffeine_vars, high_inducers, med_inducers, low_inducers){
  na_to_none <- function(x) ifelse(is.na(x),'NONE',x)
  linear_pheno <- pheno_baseline %>% dplyr::select(matches('^FID$|^IID$|_start_Drug_1'), paste0(caffeine_vars, "_caffeine"))
  linear_covar <- pheno_baseline %>% dplyr::select(matches('^FID$|^IID$|^sex$|^Age_Drug_1$|Age_sq_Drug_1$|^PC[0-9]$|^PC[1][0-9]$|^PC20|^Age_caffeine'))

  list_covar <- list(linear=linear_covar)
  list_pheno <- list(linear=linear_pheno)
  out <- numeric()
  
  for(sub in names(pheno_followup))
  {
    data_drug <- pheno_followup[[sub]]
    pheno <-  unlist(str_split(sub, "[.]"))[2]
    drug <-  unlist(str_split(sub, "[.]"))[1]
    drugs_analyzed <- test_drugs[which(test_drugs$class==drug),] %>% pull(drugs) %>% unlist()
    remove_drugs <- numeric()
    count <- 1
    for(drug_list_name in c("high_inducers", "med_inducers", "low_inducers")){
      drug_list <- get(drug_list_name)
      if(!any(drugs_analyzed %in% drug_list)){
        remove_drug_i <- drug_list
        remove_drugs <- c(remove_drugs, remove_drug_i)
      }
      if(any(drugs_analyzed %in% drug_list)){
        break
        #remove_drug_i <- drug_list[which(!drug_list %in% drugs_analyzed)]
        #remove_drugs <- c(remove_drugs, remove_drug_i)
      }
      count <- count + 1
    }

    interaction_outcome_pheno <- interaction_outcome[which(grepl(pheno, interaction_outcome))]
    
    data_drug_sensitivity <- data_drug %>% mutate(high_inducer=replace(high_inducer, high_inducer_drug_name %in% remove_drugs, NA)) %>%
      mutate_at(interaction_outcome_pheno, ~ifelse(is.na(high_inducer), NA, .)) %>%
      dplyr::select(FID, IID, high_inducer, interaction_outcome_pheno) %>% rename_at(vars(-FID,-IID),function(x) paste0(x,"_sensitivity"))

    all_var <- data_drug %>% dplyr::select(matches('^FID$|^IID$|^bmi_start$|^high_inducer$|^age_|^follow_up_time',ignore.case = F), paste0(pheno, "_start"), paste0(pheno, outcomes)) %>% #interaction_outcome) %>%
      left_join(data_drug_sensitivity) %>%
      mutate_at(c("high_inducer", "high_inducer_sensitivity"), as.character) %>%
      mutate_at(vars(high_inducer, high_inducer_sensitivity),  ~(recode(.,"0" = "NoDrug", "1" = "Drug"))) %>%
      rename_at(vars(-FID,-IID),function(x) paste0(x,"_",drug)) %>%
      mutate_if(.predicate = is.character,.funs = na_to_none)
     

    outliers_removed <- all_var %>% 
      mutate_at(.vars = c(paste0(pheno, outcomes,"_", drug), paste0(pheno, outcomes,"_sensitivity_", drug)), .funs= ~ifelse(abs(.)>mean(., na.rm=T)+5*sd(., na.rm=T), NA, .))
    
    
    NA_count_w_outliers <- all_var %>%  dplyr::select(c(paste0(pheno, outcomes,"_", drug), paste0(pheno, outcomes,"_sensitivity_", drug))) %>%
      summarise_all(~(sum(is.na(.))))
    
    NA_count_wo_outliers <- outliers_removed %>%  dplyr::select(c(paste0(pheno, outcomes,"_", drug), paste0(pheno, outcomes,"_sensitivity_", drug))) %>%
      summarise_all(~(sum(is.na(.))))
    
    NA_difference <- NA_count_wo_outliers - NA_count_w_outliers
    col_old <- colnames(NA_difference)
    col_new <- gsub(pattern = paste0(pheno, "_"),replacement = "pheno_", x  = col_old)
    col_new <- gsub(pattern = paste0("_", drug),replacement = "", x  = col_new)
    
    NA_difference <- cbind(sub, NA_difference)
    colnames(NA_difference) <- c("drug_cat", col_new)
    out <- rbind(out, NA_difference)
  }

  return(out)
}


```


```{r drake_plans, include = FALSE}


GWAS_process <- drake_plan(
  test_drugs_num = 1:dim(!!test_drugs)[1],
  NA_6mo = target(count_6mo_NAs(!!pheno_baseline_dup, !!test_drugs, test_drugs_num, !!low_inducers, !!high_inducers),
                  dynamic = map(test_drugs_num)),
  example_data = example_df(),
  num_outliers_removed = count_outliers (!!pheno_baseline_dup, !!pheno_followup_dup, !!caffeine_vars, test_drugs[1], !!high_inducers, !!med_inducers, !!low_inducers)
  
)

# make(GWAS_process)

num_outliers_removed <- count_outliers (pheno_baseline_dup, pheno_followup_dup, caffeine_vars, high_inducers, med_inducers, low_inducers)
  
loadd(NA_6mo)
loadd(example_data)
```


**The following document outlines and summarizes the GWAS analyses performed in PSYMETAB.**

# Analysis preparation.

## Sample data. 

Original data has the following structure: 

```{r sample_data, echo = FALSE}

example_data %>% dplyr::filter(Participant == "A") %>% kable() %>%
  kable_styling(full_width = F, position = "left") %>% scroll_box(width = "100%")
example_data %>% dplyr::filter(Participant == "B") %>% kable() %>%
  kable_styling(full_width = F, position = "left") %>% scroll_box(width = "100%")
example_data %>% dplyr::filter(Participant == "C") %>% kable() %>%
  kable_styling(full_width = F, position = "left") %>% scroll_box(width = "100%") 

```

Need to transform the above data to have one row per participant with the following columns:

- Drug (Yes/No),
- Outcome of interest (BMI change, slope, weighted slope, etc).
- Relevant covariates (age, sex, follow-up time, etc).

Questions to consider:

- If we are interested in drug  `XX`, which follow-up do we choose (`10001` or `10004`)? 
- If we are interested in a drug (e.g. `ZZ`) which is not taken by our patient, which follow-up do we use as control?


## Data filtering and quality control. 

Follow-ups (i.e. single rows as in sample data above) were removed if:

- Missing height data. 
- Missing weight data. 
- Height +/- 5SD outside mean (this removes patients with the following heights (cm): 106 106 106 106 106 106 106 106 116  96  96  90).
- Missing date. 
- Missing drug information (column name `AP1`).
- Multiple sexes listed for the same participant (i.e. two different follow-up listed both M and F).
- If multiple follow-ups for the same drug, consider to only round 1. 

Data corrections: 

- Height was defined as mean of all heights. 
- `Risperdal` was redefined as `Risperidone`. 
- `Paliperidone` was redefined as `Risperidone`. 
- Drugs with `depot` or `retard` appendix were considered as original drug. 
- All values equal to `999` were set to missing. 
- If any sex field was missing, sex was taken from other follow-ups (if possible).

## General settings. 

- Analyses were performed in each ethnicity seperately and then meta-anlayzed.
- Min follow-up time was set as `r min_follow_up` days (i.e. follow-up <= `r min_follow_up` days).
- Some analyses were restricted to ~6 months of followup (i.e. follow-up <= `r follow_up_limit` days).

## Drug classification. 

After discussion with others, it was decided to define drugs into three classes

1. AP1 pour faible risque de prise de poids:

Amisulpride, Aripiprazole, Brexpiprazole, Cariprazine, Carbamazepine, Chlorprothixene, Flupentixol, Fluphenazine, Haloperidol, Lurasidone,
Pipampérone, Promazine, Sertindole, Sulpiride, Tiapride.

2. AP2 pour risque intermédiaire de prise de poids:

Amitriptyline, Asenapine, Clomipramine, Dibenzepine, Doxepine, Imipramine, Lévomépromazine, Lithium, Mirtazapine,
Nortriptyline, Opipramol, Palipéridone, Quétiapine, Rispéridone, Trimipramine,  Zuclopenthixol.

3. AP3 pour risque élevé de prise de poids:

Clozapine, Olanzapine, Valproate.

Specifically, high, medium and low inducers were defined as follows: 

```{r inducer_class, echo = TRUE}
high_inducers
med_inducers
low_inducers
```

An additional group of drugs were defined as `antipsychotics` which excludes mood stabilizers and antidepressants as follows:
```{r antipsychotic_definition, echo = TRUE}
antipsychotics
```

## Follow-up selection criteria and drug (yes/no) definition

- All drugs were defined based on their propensity to induce weight gain, and fall into one of three categories (see above): 
  1. Low inducers 
  2. Medium inducers
  3. High inducers
- To test the gene*drug effect of a drug(s) of interest:
  - **Cases** (participants that have taken drug of interest) were selected as:
    - Participants who have taken the drug of interest at least once. 
    - Partcipants who have been followed up >= `min_follow_up` (`r min_follow_up` days) for drug(s) of interest.
    - If particpant has multiple follow-ups for the drug of interest that have duration >= `min_follow_up`, the first follow-up is retained.
  - **Controls** were selected as: 
    - Participants who do not meet above criteria. 
    - Only follow-ups with duration > `min_follow_up` are retained.
    - If particpant has multiple alternate follow-ups, *low inducing* drugs are prioritized, followed-by *medium inducing* drugs, and lastly *high inducing* drugs.
    - Note that all controls are followed up for an alternate drug. 
  - Individuals were filtered if they had taken the drug of interest at some point but it was not specifically followed. Therefore the follow-up data set is smaller than the baseline data set. These individuals were removed because we do not know when they took the drug and if it could have had some impact on their weight evolution. 
- In this way, a different set of cases/controls exists for each drug of interest. 
- The following drug(s) were tested: 

```{r drugs_tested2, echo = FALSE}

test_drugs %>% kable() %>%
  kable_styling(full_width = F, position = "left") 

```

- A sensitivity data-set was built for each drug set which was also tested: 
  - Define `max_category` for a set of drugs of interest as `high`, `medium`, or `low` depending on the highest inducing category found in the drug set. 
  - Remove all controls which have a follow-up for a drug **higher** than the `max_category`.
  - For example, if the gene*quetiapine interaction was tested:
    - `max_category` = `medium`
    - All cases falling in category `high` were removed. 
    - If testing a set of drugs spanning multiple categories (i.e. quetiapine, clozapine, olanzapine), the `max_category` would be `high` and no individuals would be removed in the sensitivity set. 

## Phenotype definitions.

### Linear GWAS.

- The following baseline variables were tested in a standard GWAS setting
```{r baseline_vars, echo = TRUE}
baseline_vars 
caffeine_vars
```
- If multiple follow-ups were available for a given individual, the first time-point was used. 
- Each interaction outcome (details below) * drug combination was also tested in a linear model. 
- All phenotypes had an inverse normal transformation applied (IVT) to ensure a normal distribution before downstream analyses were performed (see function below). 

```{r echo_ivt, echo = TRUE}
ivt
```

### Interaction GWAS.

- The following outcomes were tested in a gene*drug interaction GWAS setting
```{r interaction_vars, echo = TRUE}
interaction_outcome 
```
- Detailed definitions of each outcome:
  - **Pheno change**: raw change between first and last date of follow-up. 
  - **Pheno change 1mo**: raw change between first and last date of follow-up, limited to ~1 months (<= `r follow_up_1mo` days).
  - **Pheno change 3mo**: raw change between first and last date of follow-up, limited to ~3 months (<= `r follow_up_3mo` days).
  - **Pheno change 6mo**: raw change between first and last date of follow-up, limited to ~6 months (<= `r follow_up_6mo` days).
  - **Pheno slope**: beta coefficient of a linear model betwen `BMI` (y) and `days of follow-up` (x) (i.e. `date_difference_first` in example data above).
  - **Pheno slope weight**: weighted beta coefficient from model used for **Pheno slope** by SE (i.e `std_beta <- beta/se`).
  - **Pheno slope 6mo**: same as **Pheno slope** but follow-up time restricted to ~6 months (<= `r follow_up_6mo` days).
  - **Pheno slope weight 6mo**: same as **Pheno slope weight** but follow-up time restricted to ~6 months (<= `r follow_up_6mo` days).
- As above, all phenotpyes were tranformed according to IVT before downstream analyses.  
  - Note that IVT was performed on full data set instead of before sensitivity phenotypes are created
  - This means that the data used in the sensitivity models is exactly the same as the full models, just with some participants removed.
- Drug definitions are defined above.
- According to the drug definition (case/control), there could be some participants with:
  - A valid follow-up value if not limited to 6 months.
  - When limited to 6mo, the value is missing (i.e. the first follow-up visit is > `r follow_up_limit` days after the baseline visit), therefore there is no way to define BMI change or slope (as above).
  - Rather than redefine the control value, these participants simply were set to missing. 
  - The number of participants in this criteria for each drug are shown below (in other words, these participants could be redefined with a different follow-up for the 6mo period if we wanted to use a non-NA value):
```{r NA_6mo, echo = FALSE}
sapply(NA_6mo, length)

```  
- Number of models tested = 2 (original/sensitivity model) * length(drugs_tested) * length(interaction_outcome) = `r 2*dim(test_drugs)[1]*length(interaction_outcome)`.
- Values falling outside 5SD of mean were removed, the number removed  for each drug/outcome combination can be seen below: 
```{r NA_6mo_count, echo = FALSE}
num_outliers_removed %>% kable() %>%
  kable_styling(full_width = F, position = "left") 
```  

### Subgroup GWAS. 

- For each drug category tested, a GWAS was performed in each group (yes/no on drug(s)).
```{r drugs_tested, echo = FALSE}
test_drugs %>% kable() %>%
  kable_styling(full_width = F, position = "left") 
```
- All outcomes used in interaction (above) were tested as outcomes.
- Number of models tested = 2 (yes/no on drug) * 2 (original/sensitivity model) * length(drugs_tested) * length(interaction_outcome) = `r 2*2*dim(test_drugs)[1]*length(interaction_outcome)`.

## Covariate definitions.

*In all models, phenotypes were adjusted for covariates via residualization rather than standard covariate adjustemnt.* 

### Linear GWAS.

- Baseline/caffeine analyses were adjusted for PC1-20, sex, age, age^2.
- Linear models using follow-up data were adjusted for the same covariates as baseline models plus: 
  - Follow-up time.
  - Follow-up time squared.
  - BMI start.
- If 6mo variable was used, follow-up time was adjusted accordingly. 

```{r linear_covars, echo = TRUE}
standard_covars 
baseline_covars 
caffeine_covars
baseline_gwas_info %>% kable() %>%
  kable_styling(full_width = F, position = "left") %>% scroll_box(width = "100%", height="500px")
```

### Interaction GWAS.

- Interaction GWAS models were adjusted for PC1-20, sex, plus: 
  - Age started drug (varied depending on drug tested)
  - Age started drug squared.
  - Follow-up time.
  - Follow-up time squared.
  - BMI at start.
- Variable of interest: **SNP*drug_status**
  
```{r interaction_covars, echo = TRUE}
standard_covars 
interaction_gwas_info %>% kable() %>%
  kable_styling(full_width = F, position = "left") %>% scroll_box(width = "100%", height="500px")
```

### Subgroup GWAS. 

- Same adjustments as interaction GWAS. 

# Model counts and distributions.

```{r subgroup_covars, echo = FALSE}
pheno_eths <- read_table2(paste0("data/processed/phenotype_data/", study_name, "_inferred_eths.txt")) %>% mutate(FID = paste0(COUNT, "_", GPCR))

pheno <- GWAS_input$full_pheno %>% full_join(pheno_eths)
covar <- GWAS_input$full_covar %>% full_join(pheno_eths)

out <- numeric()
for(i in 1:length(drug_classes)){
  for(j in 1:length(interaction_outcome)){
    for(analysis in c("_", "_sensitivity_")){
      drug_class <- drug_classes[i]
      outcome <- interaction_outcome[j]
      outcome_colname <- paste0(outcome,analysis,drug_classes[i])
      drug_colname <- paste0("high_inducer", analysis, drug_classes[i])
      
      pheno_temp <- pheno[which(!is.na(pheno[[outcome_colname]])),] 
      
      # pheno_temp <- pheno[which(!is.na(pheno[[outcome_colname]])),] %>% 
      #   mutate_at(.vars = outcome_colname, .funs= ~ifelse(abs(.)>mean(., na.rm=T)+3*sd(., na.rm=T), NA, .)) %>% ## remove outliers
      #   
      
      drug_eth_count <- table(pheno_temp[[drug_colname]], pheno_temp[["eth"]])
      sensitivity <- ifelse(analysis=="_", "No", "Yes")
      n <- dim(pheno_temp)[1]
      row_temp <- cbind(drug_class, outcome, sensitivity, n,
                        drug_eth_count["Drug", "CEU"], drug_eth_count["NoDrug", "CEU"],
                        drug_eth_count["Drug", "MIXED"], drug_eth_count["NoDrug", "MIXED"],
                        drug_eth_count["Drug", "YRI"], drug_eth_count["NoDrug", "YRI"])
      
      out <- rbind(out, row_temp)
      
      if(j==1){
        hist(pheno_temp[[outcome_colname]], main=outcome_colname)
        #https://www.r-graph-gallery.com/histogram_several_group.html
      }
      #summary(pheno_temp[[outcome_colname]]) hist(pheno_temp[[outcome_colname]])
      
    }
  }
}

colnames(out) <- c("Drug_class", "Interaction_outcome", "Sensitivity_Model", "N",
                   "Drug_CEU", "NoDrug_CEU", 
                   "Drug_MIXED", "NoDrug_MIXED", 
                   "Drug_YRI", "NoDrug_YRI")

out <- out %>% as_tibble()
out %>% kable() %>%
  kable_styling(full_width = F, position = "left") %>% scroll_box(width = "100%", height="500px")

```


# GWAS results.

## Baseline

## Interaction

```{r interaction_images, eval=FALSE, echo=FALSE, out.width='100%'}

interaction_fig_files <- numeric()
for(i in 1:length(drug_classes)){
  for(j in 1:length(interaction_outcome)){
    for(analysis in c("_", "_sensitivity_")){
      drug_class <- drug_classes[i]
      outcome <- interaction_outcome[j]
      
      CEU_qq_file <- paste0("GWAS/interaction/processed/PSYMETAB_GWAS_", outcome, analysis, drug_class, "_CEU_qq.png")
      MIXED_qq_file <- paste0("GWAS/interaction/processed/PSYMETAB_GWAS_", outcome, "_", drug_class, "_MIXED_qq.png")
      META_qq_file <- paste0("GWAS/interaction/processed/PSYMETAB_GWAS_", outcome, "_", drug_class, "_META_qq.png")
      
      
      CEU_man_file <- paste0("GWAS/interaction/processed/PSYMETAB_GWAS_", outcome, "_", drug_class, "_CEU_manhattan.png")
      MIXED_man_file <- paste0("GWAS/interaction/processed/PSYMETAB_GWAS_", outcome, "_", drug_class, "_MIXED_manhattan.png")
      META_man_file <- paste0("GWAS/interaction/processed/PSYMETAB_GWAS_", outcome, "_", drug_class, "_META_manhattan.png")
      
      interaction_fig_files <- c(interaction_fig_files, c(CEU_qq_file, META_qq_file, CEU_man_file, META_man_file))
      
      
      #https://community.rstudio.com/t/show-png-images-consecutively-with-loop-in-r-markdown-chunk-html/45279
      
    }
  }
}

getwd()
file.exists(interaction_fig_files)

knitr::include_graphics("analysis/GWAS/interaction/processed/PSYMETAB_GWAS_bmi_change_all_CEU_qq.png")

interaction_fig_files[1:10]


knitr::include_graphics(interaction_fig_files[1:5])


img<-OpenImageR::readImage(interaction_fig_files[1])

#img<-OpenImageR::readImage("analysis/GWAS/interaction/processed/PSYMETAB_GWAS_bmi_slope_weight_valproate_MIXED_qq.png")
imageShow(img)

figures <- c("rplot.jpg", "Rlogo.png", "Rlogo2.png")
file.exists(figures)
knitr::include_graphics(figures)

#test
knitr::include_graphics("rplot.jpg")

```


## Subgroup 

```{r subgroup_images, eval=F, echo=FALSE}

fig_files <- numeric()
for(i in 1:length(drug_classes)){
  for(j in 1:length(interaction_outcome)){
    for(analysis in c("_", "_sensitivity_")){
      drug_class <- drug_classes[i]
      outcome <- interaction_outcome[j]
      
      CEU_qq_file <- paste0("../analysis/GWAS/interaction/processed/PSYMETAB_GWAS_", outcome, analysis, drug_class, "_CEU_qq.png")
      MIXED_qq_file <- paste0("../analysis/GWAS/interaction/processed/PSYMETAB_GWAS_", outcome, "_", drug_class, "_MIXED_qq.png")
      META_qq_file <- paste0("../analysis/GWAS/interaction/processed/PSYMETAB_GWAS_", outcome, "_", drug_class, "_META_qq.png")
      
      
      CEU_man_file <- paste0("../analysis/GWAS/interaction/processed/PSYMETAB_GWAS_", outcome, "_", drug_class, "_CEU_manhattan.png")
      MIXED_man_file <- paste0("../analysis/GWAS/interaction/processed/PSYMETAB_GWAS_", outcome, "_", drug_class, "_MIXED_manhattan.png")
      META_man_file <- paste0("../analysis/GWAS/interaction/processed/PSYMETAB_GWAS_", outcome, "_", drug_class, "_META_manhattan.png")
      
      fig_files <- c(fig_files, c(CEU_qq_file, META_qq_file, CEU_man_file, META_man_file))
      
      
      #https://community.rstudio.com/t/show-png-images-consecutively-with-loop-in-r-markdown-chunk-html/45279
      
    }
  }
}

knitr::include_graphics(fig_files)
  
#img<-OpenImageR::readImage("analysis/GWAS/interaction/processed/PSYMETAB_GWAS_bmi_slope_weight_valproate_MIXED_qq.png")
#imageShow(img)
```



```{r, eval=F}
het_out <-
  fread(
    paste0("analysis/GWAS/subgroup/", eth, "/", output, "_", suffix,
      "_", eth, ".", outcome_var, ".het"),
    header = F,
    data.table = F
  )

sig_list <- which(as.numeric(het_out[, 2]) < 5e-08)
length(sig_list)
colnames(het_out) <- c("ID", "P_HET")
result <- cbind(filter, "P_HET" = het_out$P_HET)
png(
  paste0("analysis/GWAS/subgroup/", eth, "/", output, "_", suffix, "_", eth, ".", outcome_var, ".het.manhattan.png"),
  width = 2000,
  height = 1000,
  pointsize = 18
)
manhattan(result, p = "P_HET", chr = "CHROM")
dev.off()

png(
  paste0(
    "analysis/GWAS/subgroup/", eth, "/", output, "_", suffix, "_", eth, ".", outcome_var, ".het.qq.png"),
  width = 2000,
  height = 1000, 
  pointsize = 18
)
qq(result$P_HET)
dev.off()


#### GWAS
    sig_nodrug <-nodrug[ which(nodrug$P < 5e-08),]
    sig_drug <- drug[which(drug$P < 5e-08),]
    for(data in c("nodrug", "drug"))
    {
      gwas_result <- get(data)
      joint <- reduce(list(gwas_result,freq,info), full_join, by = "ID")
      sig <- joint %>%
              mutate_at("P", as.numeric) %>%
              filter(P < 5e-06) %>%
              filter(ALT_FREQS > maf_threshold & ALT_FREQS < (1 - maf_threshold)) %>%
              filter(R2 > info_threshold)

      joint_maf <- joint %>% filter(ALT_FREQS > maf_threshold & ALT_FREQS < (1- maf_threshold))%>% mutate_at("P", as.numeric)
      sig <- joint_maf  %>% filter(P < gw_sig)
      png("man_interaction.png", width=2000, height=1000, pointsize=18)
      manhattan(joint_maf)
      dev.off()

      png("interaction_qq2.png", width=2000, height=1000, pointsize=18)
      qq(joint_maf$P)
      dev.off()


            qq(gwas_result2$P)
    # sig_nodrug <- sig
    }
    
    gwas_result <- fread(result, data.table=F, stringsAsFactors=F)
  gwas_result2 <- gwas_result %>% rename(CHR = "#CHROM") %>% rename(BP = POS) %>% filter(!is.na(P))


  #### GWAS
  sig_nodrug <-nodrug[ which(nodrug$P < 5e-08),]
  sig_drug <- drug[which(drug$P < 5e-08),]

  for(data in c("nodrug", "drug"))
  {
  gwas_result <- get(data)
  gwas_munge <- gwas_result %>% rename(CHR = "#CHROM") %>% rename(BP = POS) %>% filter(!is.na(P))
  joint <- reduce(list(gwas_munge,freq,info), full_join, by = "ID")
  sig <- joint %>%
          mutate_at("P", as.numeric) %>%
          filter(P < 5e-06) %>%
          filter(ALT_FREQS > maf_threshold & ALT_FREQS < (1 - maf_threshold)) %>%
          filter(R2 > info_threshold)
  # sig_nodrug <- sig
  }




```


