---
title: "Home"
site: workflowr::wflow_site
output:
  workflowr::wflow_html:
    toc: false
---

```{r, echo=FALSE}
# Define variable containing url
date <- gsub('-', '--',Sys.Date())
url <- paste0("https://img.shields.io/badge/last%20change-",date,"-brightgreen.svg")
```
[![Last-changedate](`r url`)](https://github.com/jennysjaarda/PSYMETAB)
[![minimal R version](https://img.shields.io/badge/R%3E%3D-3.5.0-brightgreen.svg)](https://cran.r-project.org/)

This [workflowr]() site contains all information and results about this research project.

Project details and overview can be found in the [About](about.html) page.

See [Data sources](data_sources.html) for details on data origins.
See [Setup](setup.html) for details on setting up the analyses for this project.


**Result shortcuts**

1. [Overview of processing in genomestudio](data_processing_in_genomestudio.html)
2. [Genotype quality control](genetic_quality_control.html) and [phenotype quality control](pheno_quality_control.html)
3. [Imputation report](GWAS.html)
4. [GWAS]()


**Useful papers**

**Notes**
  - [Creating custom cluster files](create_custom_cluster_files.html)

Powered by [drake](https://github.com/ropensci/drake) + [workflowr](https://github.com/jdblischak/workflowr)

### Acknowledgements
This work has been supported by the [Swiss National Science Foundation](http://www.snf.ch/en/funding/Pages/default.aspx).
