---
title: "Home"
site: workflowr::wflow_site
output:
  workflowr::wflow_html:
    toc: false
---

```{r, echo=FALSE}
# Define variable containing url
date <- gsub('-', '--',Sys.Date())
url <- paste0("https://img.shields.io/badge/last%20change-",date,"-brightgreen.svg")
```
[![Last-changedate](`r url`)](https://github.com/jennysjaarda/PSYMETAB)
[![minimal R version](https://img.shields.io/badge/R%3E%3D-3.5.0-brightgreen.svg)](https://cran.r-project.org/)

This website contains all information and results about this research project.

Project details and overview can be found in the [About](about.html) page.

See [Setup](setup.html) for details on setting up the project and the various analyses for this project.

See [Drake plans](plans.html) for a visual representation of analytical pipelines. 

## Useful shortcuts

1. [Data sources](data_sources.html) for details on data sources and origins.
2. [Overview of processing in genomestudio](data_processing_in_genomestudio.html).
3. [Genetic quality control](genetic_quality_control.html).
4. [Phenotypic quality control](pheno_quality_control.html).
5. [Genetic data extraction protocol](extract_genetic_data.html).
6. [GWAS methods](GWAS_method.html).
7. [GWAS results](GWAS_results.html).

Powered by [drake](https://github.com/ropensci/drake) + [workflowr](https://github.com/jdblischak/workflowr)

## Useful papers 

- [Clozapine-induced weight gain: a study in monozygotic twins and same-sex sib pairs.](https://pubmed.ncbi.nlm.nih.gov/16314759/)
- [A new genetic locus for antipsychotic-induced weight gain: A genome-wide study of first-episode psychosis patients using amisulpride (from the OPTiMiSE cohort).](https://pubmed.ncbi.nlm.nih.gov/32126890/)
- [Psychotropic drug-induced weight gain and other metabolic complications in a Swiss psychiatric population.](https://pubmed.ncbi.nlm.nih.gov/22316639/)
- [Pharmacogenetics of antipsychotic-induced weight gain: review and clinical implications.](https://www.nature.com/articles/mp2011109)

## Current Analyses 

### GWAS in PSYMETAB (our study)

- Run within various subsets. 
- Subsets were defined based on drug use, for various drug groups of interest (we had 9 groups of drugs that we analyzed).
- Phenotypes included various measures of the evolution of BMI: BMI change, BMI slope, weighted slope for SE, etc. 
- Restricted to only European subset.
- Phenotypes were residualized for: age, age^2, sex, PC1-20, follow-up time, follow-up time^2, and baseline BMI (Zoltan: we discussed this last time, but to again clarify - should follow-up time be included, it seems redundant if we are analyzing slope which already considers time between BMI measures).

### GWAS in UKBB 

- Phenotype: BMI slope.
- GWAS restricted to participants having at least 2 BMI measures restricted to unrelated, British individuals. 
- Phenotype was first residualized for: age, sex, PC1-40, and baseline BMI 

### COMPARISON BETWEEN PSYMETAB and UKBB: 

- In PSYMETAB, I took only endpoints involving slope since this would represent the best comparison to UKBB. 
- I extracted nominally significant hits in PSYMETAB (p < 5e-06) (as there were very few less with p< 5e-08).
- For these SNPs, I compared with UKBB using a standard heterogeneity statistic. 
- Limited to only GW-sig results based on het p-value. 
- Identified 68 GW-significant SNPs (i.e. SNPs that show significant difference between some PSYMETAB subset and UKBB based on a BMI slope measure, for only those SNPs in PSYMETAB that have p < 5e-06). 

1. Test top SNPs from our GWAS (within drug subgroups) with BMI in UKBB within a group defined as under the same drug as the association in our psychiatric cohort (using drug information data in UKBB). Note that in contrast to our psychiatric cohort, we don't have information for how long a UKBB participant has been taking a specific drug. 
2. To justify (1), ensure that BMI is a good proxy for change in BMI (i.e. the phenotype used in the psychiatric cohort), correlate BMI at end of follow-up with BMI slope in our psychiatric cohort.
3. For rare (0.01 < MAF < 0.05) SNPs, perform permutation test (i.e. shuffle phenotype ~1M times and perform same association: pheno ~ SNP + covars) to see if the effect found with the real data (i.e. unshuffled phenotype) is significantly larger than using the 1M shuffled phenotypes and not just some artifact of low AF.
4. Meta-analyze GWAS results across drug subgroups and use Cochran's Q test (i.e. test for heterogeneity) to determine effects differ across subgroups. 

## Acknowledgements

Primary analysis performed by: [Jenny Sjaarda](jennysjaarda@gmail.com).
Project supervisor: [Professor Chin Bin Eap](https://www.unil.ch/fbm/fr/home/menuinst/la-releve-academique/nominations--promotions/professeurs-a-a-z/d-g/eap-chin-bin.html).
Analysis support: [Professor Zoltan Kutalik](https://wp.unil.ch/sgg/people/zoltan/).

This work has been supported by the [Swiss National Science Foundation](http://www.snf.ch/en/funding/Pages/default.aspx).
