---
title: "Data quality control and processing"
site: workflowr::wflow_site
output:
  workflowr::wflow_html:
    toc: true
---

The following document outlines and summarizes the quality control and processing procedure that was followed to create a clean, imputed dataset.

## randomize_IDs.r
- Script run on CHUV computer before building GenomeStudio project.
- Creates a new csv file which was used to create a GenomeStudio project with data provided by lab in Geneva.
- Requires manual addition of header before uploading to GenomeStudio.
- Some samples were found to be duplicates (i.e. 2 samples at 2 different time points were analyzed for the same individual) and they were recoded to have ID ${ID}002.

## Pre quality control data prep
- Processes sex and ethnicity files to be used in QC scripts.
- Sex file was created according to the input specified on plink man page (FID, IID, sex [M/F]).
- Ethnicity input file to be used in R script for comparison to genetically derived ethnic groups (by snpWeights).
- Recodes ethnic groups as follows:
  - Changes French codes to English.
  - Changes missing to unknown.
  - Groups small ethnic groups to missing.
- A1 rsid conversion file was updated to remove all SNPs labeled with a [.] (see 'data/README.md').

## Quality control steps

**results are saved to `analysis/QC`**

0. Preprocessing
1. Strand alignment
2. MAF zero
3. Missingness
4. Sex check
5. Imputation (preparation and run)
6. Run and download imputation
![](assets/michigan_imputation_screenshot.PNG)
7. Check imputation
8. PLINK conversion
- Output of Michigan Impuation server is  in format chr:bp:ref:alt.
- Convenient to have SNPs in regular rsIDs for extraction, etc.
- If there is no known rsID, SNP name is left as chr:bp:ref:alt.
- Reference file for determining rsIDs can be found here: '/data/sgg2/jenny/data/dbSNP/dbSNP_SNP_list_chr${chr}.txt', which was processed according to description in '/SGG_generic/scripts/public_data.sh'.
9. Extract typed SNPs
10. Merge imputed SNPs
11. Relatedness
12. Ethnicty check and admixture estimation
13. HWE check
14. MAF check
15. Final processing

## Results

## Code

**code was run in 5 steps**

1. [code/pre_imputation_qc.sh](https://github.com/jennysjaarda/PSYMETAB/blob/master/code/pre_imputation_qc.sh)
2. [code/download_imputation.sh](https://github.com/jennysjaarda/PSYMETAB/blob/master/code/pre_imputation_qc.sh)
3. [code/check_imputation.sh](https://github.com/jennysjaarda/PSYMETAB/blob/master/code/check_imputation.sh)
4. [code/post_imputation_qc.sh](https://github.com/jennysjaarda/PSYMETAB/blob/master/code/post_imputation_qc.sh)
5. [code/final_processing.sh](https://github.com/jennysjaarda/PSYMETAB/blob/master/code/final_processing.sh)
